<!DOCTYPE html><html class=writer-html5 lang=en> <head><meta charset=utf-8><meta http-equiv=X-UA-Compatible content="IE=edge"><meta name=viewport content="width=device-width, initial-scale=1.0"><meta name=author content=CCBR><link rel="shortcut icon" href=../../img/favicon.ico><title>Graphical Interface - XAVIER Documentation</title><link rel=stylesheet href=../../css/theme.css><link rel=stylesheet href=../../css/theme_extra.css><link rel=stylesheet href=https://cdnjs.cloudflare.com/ajax/libs/highlight.js/11.8.0/styles/github.min.css><link href=../../css/custom.css rel=stylesheet><link href=../../css/version-select.css rel=stylesheet><script>
        // Current page data
        var mkdocs_page_name = "Graphical Interface";
        var mkdocs_page_input_path = "usage/gui.md";
        var mkdocs_page_url = null;
      </script><!--[if lt IE 9]>
      <script src="../../js/html5shiv.min.js"></script>
    <![endif]--><script src=https://cdnjs.cloudflare.com/ajax/libs/highlight.js/11.8.0/highlight.min.js></script><script>hljs.highlightAll();</script></head> <body class=wy-body-for-nav role=document> <div class=wy-grid-for-nav> <nav data-toggle=wy-nav-shift class="wy-nav-side stickynav"> <div class=wy-side-scroll> <div class=wy-side-nav-search> <a href=../.. class="icon icon-home"> XAVIER Documentation </a><div role=search> <form id=rtd-search-form class=wy-form action=../../search.html method=get> <input type=text name=q placeholder="Search docs" aria-label="Search docs" title="Type search term here"> </form> </div> </div> <div class="wy-menu wy-menu-vertical" data-spy=affix role=navigation aria-label="Navigation menu"> <ul> <li class=toctree-l1><a class="reference internal" href=../..>About</a> </li> </ul> <p class=caption><span class=caption-text>Command Line</span></p> <ul> <li class=toctree-l1><a href=../run/ class="reference internal">xavier run</a> </li> <li class=toctree-l1><a href=../unlock/ class="reference internal">xavier unlock</a> </li> <li class=toctree-l1><a href=../cache/ class="reference internal">xavier cache</a> </li> </ul> <ul class=current> <li class="toctree-l1 current"><a class="reference internal current" href=#>Graphical Interface</a> <ul class=current> <li class=toctree-l2><a class="reference internal" href=#1-synopsis>1. Synopsis</a> </li> <li class=toctree-l2><a class="reference internal" href=#2-setting-up-xavier>2. Setting up XAVIER</a> <ul> <li class=toctree-l3><a class="reference internal" href=#21-login-to-cluster>2.1 Login to cluster</a> </li> <li class=toctree-l3><a class="reference internal" href=#22-grab-an-interactive-node>2.2 Grab an interactive node</a> </li> <li class=toctree-l3><a class="reference internal" href=#23-load-ccbrpipeliner-module>2.3 Load ccbrpipeliner module</a> </li> </ul> </li> <li class=toctree-l2><a class="reference internal" href=#3-running-xavier>3. Running XAVIER</a> <ul> <li class=toctree-l3><a class="reference internal" href=#31-launching-xavier-gui>3.1 Launching XAVIER GUI</a> </li> <li class=toctree-l3><a class="reference internal" href=#32-folder-paths-and-reference-genomes>3.2 Folder paths and reference genomes</a> </li> <li class=toctree-l3><a class="reference internal" href=#33-analysis-mode>3.3 Analysis mode</a> <ul> <li class=toctree-l4><a class="reference internal" href=#33a-tumor-normal-pair-analysis>3.3a Tumor-normal pair analysis</a> </li> <li class=toctree-l4><a class="reference internal" href=#33b-tumor-only-analysis>3.3b Tumor-only analysis</a> </li> </ul> </li> <li class=toctree-l3><a class="reference internal" href=#34-submit-xavier-job>3.4 Submit XAVIER job</a> </li> <li class=toctree-l3><a class="reference internal" href=#35-additional-settings>3.5 Additional settings</a> </li> </ul> </li> <li class=toctree-l2><a class="reference internal" href=#4-special-instructions-for-biowulf>4. Special instructions for Biowulf</a> </li> </ul> </li> </ul> <p class=caption><span class=caption-text>Pipeline Details</span></p> <ul> <li class=toctree-l1><a href=../../pipeline-details/overview/ class="reference internal">Overview</a> </li> <li class=toctree-l1><a href=../../pipeline-details/methods/ class="reference internal">Methods</a> </li> <li class=toctree-l1><a href=../../pipeline-details/tools/ class="reference internal">Tools</a> </li> <li class=toctree-l1><a href=../../pipeline-details/settings/ class="reference internal">Settings</a> </li> <li class=toctree-l1><a href=../../pipeline-details/output/ class="reference internal">Output Files</a> </li> </ul> <p class=caption><span class=caption-text>FAQ</span></p> <ul> <li class=toctree-l1><a href=../../contributing/ class="reference internal">How to contribute</a> </li> </ul> <ul> <li class=toctree-l1><a href=../../license/ class="reference internal">License</a> </li> </ul> </div> </div> </nav> <section data-toggle=wy-nav-shift class=wy-nav-content-wrap> <nav class=wy-nav-top role=navigation aria-label="Mobile navigation menu"> <i data-toggle=wy-nav-top class="fa fa-bars"></i> <a href=../..>XAVIER Documentation</a> </nav> <div class=wy-nav-content> <div class=rst-content><div role=navigation aria-label="breadcrumbs navigation"> <ul class=wy-breadcrumbs> <li><a href=../.. class="icon icon-home" aria-label=Docs></a></li> <li class="breadcrumb-item active">Graphical Interface</li> <li class=wy-breadcrumbs-aside> <a href=https://github.com/CCBR/XAVIER/edit/unified/docs/usage/gui.md>Edit on CCBR/XAVIER</a> </li> </ul> <hr> </div> <div role=main class=document itemscope=itemscope itemtype=http://schema.org/Article> <div class=section itemprop=articleBody> <h1 id=getting-started>Getting started<a class=headerlink href=#getting-started title="Permanent link">&para;</a></h1> <h2 id=1-synopsis>1. Synopsis<a class=headerlink href=#1-synopsis title="Permanent link">&para;</a></h2> <p>XAVIER pipeline can be executed from either using the graphical user interface (GUI) or the command line interface (CLI). GUI offers a more interactive way for the user to provide input and adjust parameter settings. This part of the documentation describes how to run xavier using the GUI (with screenshots). See <a href=../run/ >Command Line</a> tab to read more about the <code>xavier</code> executable and running XAVIER pipeline using the CLI.</p> <h2 id=2-setting-up-xavier>2. Setting up XAVIER<a class=headerlink href=#2-setting-up-xavier title="Permanent link">&para;</a></h2> <h3 id=21-login-to-cluster>2.1 Login to cluster<a class=headerlink href=#21-login-to-cluster title="Permanent link">&para;</a></h3> <div class=highlight><pre><span></span><code><span class=c1># Setup Step 1.) ssh into cluster&#39;s head node</span>
<span class=c1># example below for Biowulf cluster</span>
ssh<span class=w> </span>-Y<span class=w> </span><span class=nv>$USER</span>@biowulf.nih.gov
</code></pre></div> <h3 id=22-grab-an-interactive-node>2.2 Grab an interactive node<a class=headerlink href=#22-grab-an-interactive-node title="Permanent link">&para;</a></h3> <div class=highlight><pre><span></span><code><span class=c1># Setup Step 2.) Please do not run XAVIER on the head node!</span>
<span class=c1># Grab an interactive node first</span>
sinteractive<span class=w> </span>--tunnel<span class=w>  </span>--time<span class=o>=</span><span class=m>12</span>:00:00<span class=w> </span>--mem<span class=o>=</span>8gb<span class=w>  </span>--cpus-per-task<span class=o>=</span><span class=m>4</span>
</code></pre></div> <h3 id=23-load-ccbrpipeliner-module>2.3 Load <code>ccbrpipeliner</code> module<a class=headerlink href=#23-load-ccbrpipeliner-module title="Permanent link">&para;</a></h3> <p><em>NOTE: <code>ccbrpipeliner</code> is a custom module created on biowulf which contains various NGS data analysis pipelines developed, tested, and benchmarked by experts at CCBR.</em></p> <div class=highlight><pre><span></span><code><span class=c1># Setup Step 3.) Add ccbrpipeliner module</span>
module<span class=w> </span>purge<span class=w> </span><span class=c1># to reset the module environment</span>
module<span class=w> </span>load<span class=w> </span>ccbrpipeliner
</code></pre></div> <p>If the module was loaded correctly, the greetings message should be displayed.</p> <div class=highlight><pre><span></span><code><span class=o>[</span>+<span class=o>]</span><span class=w> </span>Loading<span class=w> </span>ccbrpipeliner<span class=w>  </span><span class=m>5</span><span class=w>  </span>...
<span class=c1>###########################################################################</span>
<span class=w>                                </span>CCBR<span class=w> </span>Pipeliner
<span class=c1>###########################################################################</span>
<span class=w>    </span><span class=s2>&quot;ccbrpipeliner&quot;</span><span class=w> </span>is<span class=w> </span>a<span class=w> </span>suite<span class=w> </span>of<span class=w> </span>end-to-end<span class=w> </span>pipelines<span class=w> </span>and<span class=w> </span>tools
<span class=w>    </span>Visit<span class=w> </span>https://github.com/ccbr<span class=w> </span><span class=k>for</span><span class=w> </span>more<span class=w> </span>details.
<span class=w>    </span>Pipelines<span class=w> </span>are<span class=w> </span>available<span class=w> </span>on<span class=w> </span>BIOWULF<span class=w> </span>and<span class=w> </span>FRCE.
<span class=w>    </span>Tools<span class=w> </span>are<span class=w> </span>available<span class=w> </span>on<span class=w> </span>BIOWULF,<span class=w> </span>HELIX<span class=w> </span>and<span class=w> </span>FRCE.

<span class=w>    </span>The<span class=w> </span>following<span class=w> </span>pipelines/tools<span class=w> </span>will<span class=w> </span>be<span class=w> </span>loaded<span class=w> </span><span class=k>in</span><span class=w> </span>this<span class=w> </span>module:

<span class=w>    </span>RENEE<span class=w> </span>v2.5<span class=w> </span>https://ccbr.github.io/RENEE/
<span class=w>    </span>XAVIER<span class=w> </span>v3.0<span class=w> </span>https://ccbr.github.io/XAVIER/
<span class=w>    </span>CARLISLE<span class=w> </span>v2.4<span class=w> </span>https://ccbr.github.io/CARLISLE/
<span class=w>    </span>CHAMPAGNE<span class=w> </span>v0.2<span class=w> </span>https://ccbr.github.io/CHAMPAGNE/
<span class=w>    </span>CRUISE<span class=w> </span>v0.1<span class=w> </span>https://ccbr.github.io/CRUISE/

<span class=w>    </span>spacesavers2<span class=w> </span>v0.10<span class=w> </span>https://ccbr.github.io/spacesavers2/
<span class=w>    </span>permfix<span class=w> </span>v0.6<span class=w> </span>https://github.com/ccbr/permfix
<span class=c1>###########################################################################</span>
Thank<span class=w> </span>you<span class=w> </span><span class=k>for</span><span class=w> </span>using<span class=w> </span>CCBR<span class=w> </span>Pipeliner
<span class=c1>###########################################################################</span>
</code></pre></div> <p>To check the current version of XAVIER, enter:</p> <div class=highlight><pre><span></span><code>xavier<span class=w> </span>--version
</code></pre></div> <h2 id=3-running-xavier>3. Running XAVIER<a class=headerlink href=#3-running-xavier title="Permanent link">&para;</a></h2> <h3 id=31-launching-xavier-gui>3.1 Launching XAVIER GUI<a class=headerlink href=#31-launching-xavier-gui title="Permanent link">&para;</a></h3> <p>To run the XAVIER pipeline from the GUI, simply enter:</p> <div class=highlight><pre><span></span><code>xavier<span class=w> </span>gui
</code></pre></div> <p>and it will launch the XAVIER window.</p> <blockquote> <p>Note: Please wait until <code>window created!</code> message appears on the terminal.</p> </blockquote> <p><img alt=xavier_launchscreen src=../images/gui_launch.png></p> <h3 id=32-folder-paths-and-reference-genomes>3.2 Folder paths and reference genomes<a class=headerlink href=#32-folder-paths-and-reference-genomes title="Permanent link">&para;</a></h3> <p>To enter the location of the input folder containing FASTQ files and the location where the output folders should be created, either simply type the absolute paths</p> <p><img alt=xavier_enterPath src=../images/gui_path.png></p> <p>or use the <strong>Browse</strong> tab to choose the input and output directories</p> <p><img alt=xavier_browsePath src=../images/gui_browse.png></p> <p>Next, from the drop down menu select the reference genome (hg38/mm10)</p> <p><img alt=xavier_genome src=../images/gui_genome.png></p> <p>and enter a job name of this run.</p> <p><img alt=xavier_jobname src=../images/gui_jobname.png></p> <h3 id=33-analysis-mode>3.3 Analysis mode<a class=headerlink href=#33-analysis-mode title="Permanent link">&para;</a></h3> <p>XAVIER pipeline can be run in two different modes:\ (A) Tumor-normal pair \ (B) Tumor-only</p> <h4 id=33a-tumor-normal-pair-analysis>3.3a Tumor-normal pair analysis<a class=headerlink href=#33a-tumor-normal-pair-analysis title="Permanent link">&para;</a></h4> <p>In case of tumor-normal pairs, a tab-delimited text file is neeed that contains the list of normal and tumor samples. For example,</p> <div class=highlight><pre><span></span><code>Normal<span class=w>  </span>Tumor
sample1-normal<span class=w>     </span>sample1-tumor
sample2-normal<span class=w>     </span>sample2-tumor
sample3-normal<span class=w>     </span>sample3-tumor
sample4-normal<span class=w>     </span>sample4-tumor
</code></pre></div> <p>Similar to input and output folder paths, either type the path to the pairsInfo.txt file or use the <strong>Browse</strong> tab.</p> <p>In case of paired mode, XAVIER can also perform copy number variants (CNV) analysis.</p> <p><img alt=xavier_pair src=../images/gui_pair.png></p> <h4 id=33b-tumor-only-analysis>3.3b Tumor-only analysis<a class=headerlink href=#33b-tumor-only-analysis title="Permanent link">&para;</a></h4> <p>In case the paired normal samples are unavailable, XAVIER pipeline can be run in tumor-only mode which does not require paired samples information. However, in the absence of matching normal samples, CNV analysis is also unavailable.</p> <p><img alt=xavier_tumorOnly src=../images/gui_tumorOnly.png></p> <h3 id=34-submit-xavier-job>3.4 Submit XAVIER job<a class=headerlink href=#34-submit-xavier-job title="Permanent link">&para;</a></h3> <p>After all the information is filled out, press <strong>Submit</strong>.</p> <p>If the pipeline detects no errors and the run was submitted, a new window appears that has the output of a "dry-run" which summarizes each step of the pipeline.</p> <p><img alt=xavier_dryrun src=../images/gui_dryrun.png></p> <p>Click <strong>OK</strong></p> <p>A dialogue box will popup to confirm submitting the job to slurm.</p> <p><img alt=xavier_submit src=../images/gui_submit.png></p> <p>Click <strong>Yes</strong></p> <p>The dry-run output will be displayed again and the master job will be submitted. An email notification will be sent out when the pipeline starts and ends.</p> <p>The XAVIER gui will ask to submit another job.</p> <p><img alt=xavier_resubmit src=../images/gui_resubmit.png></p> <p>Click <strong>Yes</strong> to start again or <strong>No</strong> to close the XAVIER gui.</p> <h3 id=35-additional-settings>3.5 Additional settings<a class=headerlink href=#35-additional-settings title="Permanent link">&para;</a></h3> <p>Users can input certain additional settings for the pipeline run including running an additional step to correct strand orientation bias in Formalin-Fixed Paraffin-Embedded (FFPE) samples and to provide a custom exome targets BED file. This file can be obtained from the manufacturer of the target capture kit that was used.</p> <p><img alt=gui_additionalSettings1 src=../images/gui_additionalSettings1.png></p> <p><img alt=gui_additionalSettings2 src=../images/gui_additionalSettings2.png></p> <h2 id=4-special-instructions-for-biowulf>4. Special instructions for Biowulf<a class=headerlink href=#4-special-instructions-for-biowulf title="Permanent link">&para;</a></h2> <p>XAVIER GUI natively uses the X11 Window System to run XAVIER pipeline and display the graphics on a personal desktop or laptop. However, if running XAVIER specifically on NIH's Biowulf cluster, the HPC staff recommends NoMachine (NX) to run graphics applications.</p> <p>Please see details <a href=https://hpc.nih.gov/docs/nx.html>here</a> on how to install and connect to Biowulf on your local computer using NoMachine.</p> <p>Once connected to Biowulf using NX, right click to open a terminal connection</p> <p><img alt=gui_nx_config1 src=../images/gui_nx_config1.png></p> <p>and start an interactive session.</p> <p><img alt=gui_nx_config2 src=../images/gui_nx_config2.png></p> <p>Similar to the instructions above, load <code>ccbrpipeliner</code> module and enter <code>xavier gui</code> to launch the XAVIER gui.</p> <p><img alt=gui_nx_xavier src=../images/gui_nx_xavier.png></p> </div> </div><script async src="https://www.googletagmanager.com/gtag/js?id=G-D3SL9V30KL"></script> <script>
  window.dataLayer = window.dataLayer || [];
  function gtag(){dataLayer.push(arguments);}
  gtag('js', new Date());
  gtag('config', 'G-D3SL9V30KL'); // Replace with your tracking ID
</script> </div> </div> </section> </div> <div class=rst-versions role=note aria-label=Versions> <span class=rst-current-version data-toggle=rst-current-version> <span> <a href=https://github.com/CCBR/XAVIER class="fa fa-code-fork" style="color: #fcfcfc"> CCBR/XAVIER</a> </span> <span><a href=../cache/ style="color: #fcfcfc">&laquo; Previous</a></span> <span><a href=../../pipeline-details/overview/ style="color: #fcfcfc">Next &raquo;</a></span> </span> </div> <script src=../../js/jquery-3.6.0.min.js></script> <script>var base_url = "../..";</script> <script src=../../js/theme_extra.js></script> <script src=../../js/theme.js></script> <script src=../../js/custom.js></script> <script src=../../search/main.js></script> <script src=../../js/version-select.js></script> <script>
        jQuery(function () {
            SphinxRtdTheme.Navigation.enable(true);
        });
    </script> </body> </html> 