<!doctype html><html lang=en class=no-js> <head><meta charset=utf-8><meta name=viewport content="width=device-width,initial-scale=1"><meta name=description content="XAVIER is an open-source, reproducible, and scalable best practices pipeline for analyzing Whole Exome DNA-sequencing data."><meta name=author content=CCBR><link rel=prev href=../..><link href=../unlock/ rel=next><link rel=icon href=../../assets/favicon/favicon.ico><meta name=generator content="mkdocs-1.5.3, mkdocs-material-9.5.6"><title>xavier run - XAVIER Documentation</title><link rel=stylesheet href=../../assets/stylesheets/main.50c56a3b.min.css><link rel=stylesheet href=../../assets/stylesheets/palette.06af60db.min.css><link rel=preconnect href=https://fonts.gstatic.com crossorigin><link rel=stylesheet href="https://fonts.googleapis.com/css?family=Roboto:300,300i,400,400i,700,700i%7CRoboto+Mono:400,400i,700,700i&display=fallback"><style>:root{--md-text-font:"Roboto";--md-code-font:"Roboto Mono"}</style><link rel=stylesheet href=../../css/extra.css><script>__md_scope=new URL("../..",location),__md_hash=e=>[...e].reduce((e,_)=>(e<<5)-e+_.charCodeAt(0),0),__md_get=(e,_=localStorage,t=__md_scope)=>JSON.parse(_.getItem(t.pathname+"."+e)),__md_set=(e,_,t=localStorage,a=__md_scope)=>{try{t.setItem(a.pathname+"."+e,JSON.stringify(_))}catch(e){}}</script></head> <body dir=ltr data-md-color-scheme=default data-md-color-primary=indigo data-md-color-accent=indigo> <input class=md-toggle data-md-toggle=drawer type=checkbox id=__drawer autocomplete=off> <input class=md-toggle data-md-toggle=search type=checkbox id=__search autocomplete=off> <label class=md-overlay for=__drawer></label> <div data-md-component=skip> <a href=#xavier-run class=md-skip> Skip to content </a> </div> <div data-md-component=announce> </div> <div data-md-color-scheme=default data-md-component=outdated hidden> </div> <header class=md-header data-md-component=header> <nav class="md-header__inner md-grid" aria-label=Header> <a href=../.. title="XAVIER Documentation" class="md-header__button md-logo" aria-label="XAVIER Documentation" data-md-component=logo> <img src=../../assets/icons/doc-book.svg alt=logo> </a> <label class="md-header__button md-icon" for=__drawer> <svg xmlns=http://www.w3.org/2000/svg viewbox="0 0 24 24"><path d="M3 6h18v2H3V6m0 5h18v2H3v-2m0 5h18v2H3v-2Z"/></svg> </label> <div class=md-header__title data-md-component=header-title> <div class=md-header__ellipsis> <div class=md-header__topic> <span class=md-ellipsis> XAVIER Documentation </span> </div> <div class=md-header__topic data-md-component=header-topic> <span class=md-ellipsis> xavier run </span> </div> </div> </div> <form class=md-header__option data-md-component=palette> <input class=md-option data-md-color-media data-md-color-scheme=default data-md-color-primary=indigo data-md-color-accent=indigo aria-label="Switch to dark mode" type=radio name=__palette id=__palette_0> <label class="md-header__button md-icon" title="Switch to dark mode" for=__palette_1 hidden> <svg xmlns=http://www.w3.org/2000/svg viewbox="0 0 24 24"><path d="M17 6H7c-3.31 0-6 2.69-6 6s2.69 6 6 6h10c3.31 0 6-2.69 6-6s-2.69-6-6-6zm0 10H7c-2.21 0-4-1.79-4-4s1.79-4 4-4h10c2.21 0 4 1.79 4 4s-1.79 4-4 4zM7 9c-1.66 0-3 1.34-3 3s1.34 3 3 3 3-1.34 3-3-1.34-3-3-3z"/></svg> </label> <input class=md-option data-md-color-media data-md-color-scheme=slate data-md-color-primary=red data-md-color-accent=red aria-label="Switch to light mode" type=radio name=__palette id=__palette_1> <label class="md-header__button md-icon" title="Switch to light mode" for=__palette_0 hidden> <svg xmlns=http://www.w3.org/2000/svg viewbox="0 0 24 24"><path d="M17 7H7a5 5 0 0 0-5 5 5 5 0 0 0 5 5h10a5 5 0 0 0 5-5 5 5 0 0 0-5-5m0 8a3 3 0 0 1-3-3 3 3 0 0 1 3-3 3 3 0 0 1 3 3 3 3 0 0 1-3 3Z"/></svg> </label> </form> <script>var media,input,key,value,palette=__md_get("__palette");if(palette&&palette.color){"(prefers-color-scheme)"===palette.color.media&&(media=matchMedia("(prefers-color-scheme: light)"),input=document.querySelector(media.matches?"[data-md-color-media='(prefers-color-scheme: light)']":"[data-md-color-media='(prefers-color-scheme: dark)']"),palette.color.media=input.getAttribute("data-md-color-media"),palette.color.scheme=input.getAttribute("data-md-color-scheme"),palette.color.primary=input.getAttribute("data-md-color-primary"),palette.color.accent=input.getAttribute("data-md-color-accent"));for([key,value]of Object.entries(palette.color))document.body.setAttribute("data-md-color-"+key,value)}</script> <label class="md-header__button md-icon" for=__search> <svg xmlns=http://www.w3.org/2000/svg viewbox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.516 6.516 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5Z"/></svg> </label> <div class=md-search data-md-component=search role=dialog> <label class=md-search__overlay for=__search></label> <div class=md-search__inner role=search> <form class=md-search__form name=search> <input type=text class=md-search__input name=query aria-label=Search placeholder=Search autocapitalize=off autocorrect=off autocomplete=off spellcheck=false data-md-component=search-query required> <label class="md-search__icon md-icon" for=__search> <svg xmlns=http://www.w3.org/2000/svg viewbox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.516 6.516 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5Z"/></svg> <svg xmlns=http://www.w3.org/2000/svg viewbox="0 0 24 24"><path d="M20 11v2H8l5.5 5.5-1.42 1.42L4.16 12l7.92-7.92L13.5 5.5 8 11h12Z"/></svg> </label> <nav class=md-search__options aria-label=Search> <button type=reset class="md-search__icon md-icon" title=Clear aria-label=Clear tabindex=-1> <svg xmlns=http://www.w3.org/2000/svg viewbox="0 0 24 24"><path d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12 19 6.41Z"/></svg> </button> </nav> </form> <div class=md-search__output> <div class=md-search__scrollwrap data-md-scrollfix> <div class=md-search-result data-md-component=search-result> <div class=md-search-result__meta> Initializing search </div> <ol class=md-search-result__list role=presentation></ol> </div> </div> </div> </div> </div> <div class=md-header__source> <a href=https://github.com/CCBR/XAVIER title="Go to repository" class=md-source data-md-component=source> <div class="md-source__icon md-icon"> <svg xmlns=http://www.w3.org/2000/svg viewbox="0 0 448 512"><!-- Font Awesome Free 6.5.1 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2023 Fonticons, Inc.--><path d="M439.55 236.05 244 40.45a28.87 28.87 0 0 0-40.81 0l-40.66 40.63 51.52 51.52c27.06-9.14 52.68 16.77 43.39 43.68l49.66 49.66c34.23-11.8 61.18 31 35.47 56.69-26.49 26.49-70.21-2.87-56-37.34L240.22 199v121.85c25.3 12.54 22.26 41.85 9.08 55a34.34 34.34 0 0 1-48.55 0c-17.57-17.6-11.07-46.91 11.25-56v-123c-20.8-8.51-24.6-30.74-18.64-45L142.57 101 8.45 235.14a28.86 28.86 0 0 0 0 40.81l195.61 195.6a28.86 28.86 0 0 0 40.8 0l194.69-194.69a28.86 28.86 0 0 0 0-40.81z"/></svg> </div> <div class=md-source__repository> CCBR/XAVIER </div> </a> </div> </nav> </header> <div class=md-container data-md-component=container> <nav class=md-tabs aria-label=Tabs data-md-component=tabs> <div class=md-grid> <ul class=md-tabs__list> <li class=md-tabs__item> <a href=../.. class=md-tabs__link> About </a> </li> <li class="md-tabs__item md-tabs__item--active"> <a href=./ class=md-tabs__link> Command Line </a> </li> <li class=md-tabs__item> <a href=../gui/ class=md-tabs__link> Graphical Interface </a> </li> <li class=md-tabs__item> <a href=../../pipeline-details/overview/ class=md-tabs__link> Pipeline Details </a> </li> <li class=md-tabs__item> <a href=../../faq/questions/ class=md-tabs__link> FAQ </a> </li> <li class=md-tabs__item> <a href=../../license/ class=md-tabs__link> License </a> </li> </ul> </div> </nav> <main class=md-main data-md-component=main> <div class="md-main__inner md-grid"> <div class="md-sidebar md-sidebar--primary" data-md-component=sidebar data-md-type=navigation> <div class=md-sidebar__scrollwrap> <div class=md-sidebar__inner> <nav class="md-nav md-nav--primary md-nav--lifted md-nav--integrated" aria-label=Navigation data-md-level=0> <label class=md-nav__title for=__drawer> <a href=../.. title="XAVIER Documentation" class="md-nav__button md-logo" aria-label="XAVIER Documentation" data-md-component=logo> <img src=../../assets/icons/doc-book.svg alt=logo> </a> XAVIER Documentation </label> <div class=md-nav__source> <a href=https://github.com/CCBR/XAVIER title="Go to repository" class=md-source data-md-component=source> <div class="md-source__icon md-icon"> <svg xmlns=http://www.w3.org/2000/svg viewbox="0 0 448 512"><!-- Font Awesome Free 6.5.1 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2023 Fonticons, Inc.--><path d="M439.55 236.05 244 40.45a28.87 28.87 0 0 0-40.81 0l-40.66 40.63 51.52 51.52c27.06-9.14 52.68 16.77 43.39 43.68l49.66 49.66c34.23-11.8 61.18 31 35.47 56.69-26.49 26.49-70.21-2.87-56-37.34L240.22 199v121.85c25.3 12.54 22.26 41.85 9.08 55a34.34 34.34 0 0 1-48.55 0c-17.57-17.6-11.07-46.91 11.25-56v-123c-20.8-8.51-24.6-30.74-18.64-45L142.57 101 8.45 235.14a28.86 28.86 0 0 0 0 40.81l195.61 195.6a28.86 28.86 0 0 0 40.8 0l194.69-194.69a28.86 28.86 0 0 0 0-40.81z"/></svg> </div> <div class=md-source__repository> CCBR/XAVIER </div> </a> </div> <ul class=md-nav__list data-md-scrollfix> <li class=md-nav__item> <a href=../.. class=md-nav__link> <span class=md-ellipsis> About </span> </a> </li> <li class="md-nav__item md-nav__item--active md-nav__item--section md-nav__item--nested"> <input class="md-nav__toggle md-toggle " type=checkbox id=__nav_2 checked> <label class=md-nav__link for=__nav_2 id=__nav_2_label tabindex> <span class=md-ellipsis> Command Line </span> <span class="md-nav__icon md-icon"></span> </label> <nav class=md-nav data-md-level=1 aria-labelledby=__nav_2_label aria-expanded=true> <label class=md-nav__title for=__nav_2> <span class="md-nav__icon md-icon"></span> Command Line </label> <ul class=md-nav__list data-md-scrollfix> <li class="md-nav__item md-nav__item--active"> <input class="md-nav__toggle md-toggle" type=checkbox id=__toc> <label class="md-nav__link md-nav__link--active" for=__toc> <span class=md-ellipsis> xavier run </span> <span class="md-nav__icon md-icon"></span> </label> <a href=./ class="md-nav__link md-nav__link--active"> <span class=md-ellipsis> xavier run </span> </a> <nav class="md-nav md-nav--secondary" aria-label="Table of contents"> <label class=md-nav__title for=__toc> <span class="md-nav__icon md-icon"></span> Table of contents </label> <ul class=md-nav__list data-md-component=toc data-md-scrollfix> <li class=md-nav__item> <a href=#1-about class=md-nav__link> <span class=md-ellipsis> 1. About </span> </a> </li> <li class=md-nav__item> <a href=#2-synopsis class=md-nav__link> <span class=md-ellipsis> 2. Synopsis </span> </a> <nav class=md-nav aria-label="2. Synopsis"> <ul class=md-nav__list> <li class=md-nav__item> <a href=#21-required-arguments class=md-nav__link> <span class=md-ellipsis> 2.1 Required Arguments </span> </a> </li> <li class=md-nav__item> <a href=#22-options class=md-nav__link> <span class=md-ellipsis> 2.2 Options </span> </a> </li> </ul> </nav> </li> <li class=md-nav__item> <a href=#3-example class=md-nav__link> <span class=md-ellipsis> 3. Example </span> </a> </li> </ul> </nav> </li> <li class=md-nav__item> <a href=../unlock/ class=md-nav__link> <span class=md-ellipsis> xavier unlock </span> </a> </li> <li class=md-nav__item> <a href=../cache/ class=md-nav__link> <span class=md-ellipsis> xavier cache </span> </a> </li> </ul> </nav> </li> <li class=md-nav__item> <a href=../gui/ class=md-nav__link> <span class=md-ellipsis> Graphical Interface </span> </a> </li> <li class="md-nav__item md-nav__item--section md-nav__item--nested"> <input class="md-nav__toggle md-toggle " type=checkbox id=__nav_4> <label class=md-nav__link for=__nav_4 id=__nav_4_label tabindex> <span class=md-ellipsis> Pipeline Details </span> <span class="md-nav__icon md-icon"></span> </label> <nav class=md-nav data-md-level=1 aria-labelledby=__nav_4_label aria-expanded=false> <label class=md-nav__title for=__nav_4> <span class="md-nav__icon md-icon"></span> Pipeline Details </label> <ul class=md-nav__list data-md-scrollfix> <li class=md-nav__item> <a href=../../pipeline-details/overview/ class=md-nav__link> <span class=md-ellipsis> Overview </span> </a> </li> <li class=md-nav__item> <a href=../../pipeline-details/methods/ class=md-nav__link> <span class=md-ellipsis> Methods </span> </a> </li> <li class=md-nav__item> <a href=../../pipeline-details/tools/ class=md-nav__link> <span class=md-ellipsis> Tools </span> </a> </li> <li class=md-nav__item> <a href=../../pipeline-details/settings/ class=md-nav__link> <span class=md-ellipsis> Settings </span> </a> </li> <li class=md-nav__item> <a href=../../pipeline-details/output/ class=md-nav__link> <span class=md-ellipsis> Output Files </span> </a> </li> </ul> </nav> </li> <li class="md-nav__item md-nav__item--section md-nav__item--nested"> <input class="md-nav__toggle md-toggle " type=checkbox id=__nav_5> <label class=md-nav__link for=__nav_5 id=__nav_5_label tabindex> <span class=md-ellipsis> FAQ </span> <span class="md-nav__icon md-icon"></span> </label> <nav class=md-nav data-md-level=1 aria-labelledby=__nav_5_label aria-expanded=false> <label class=md-nav__title for=__nav_5> <span class="md-nav__icon md-icon"></span> FAQ </label> <ul class=md-nav__list data-md-scrollfix> <li class=md-nav__item> <a href=../../faq/questions/ class=md-nav__link> <span class=md-ellipsis> General Questions </span> </a> </li> <li class=md-nav__item> <a href=../../contributing/ class=md-nav__link> <span class=md-ellipsis> How to contribute </span> </a> </li> </ul> </nav> </li> <li class=md-nav__item> <a href=../../license/ class=md-nav__link> <span class=md-ellipsis> License </span> </a> </li> </ul> </nav> </div> </div> </div> <div class=md-content data-md-component=content> <article class="md-content__inner md-typeset"> <h1 id=xavier-run><code>xavier <b>run</b></code><a class=headerlink href=#xavier-run title="Permanent link">&para;</a></h1> <h2 id=1-about>1. About<a class=headerlink href=#1-about title="Permanent link">&para;</a></h2> <p>The <code>xavier</code> executable is composed of several inter-related sub commands. Please see <code>xavier -h</code> for all available options.</p> <p>This part of the documentation describes options and concepts for <code>xavier <b>run</b></code> sub command in more detail. With minimal configuration, the <strong><code>run</code></strong> sub command enables you to start running xavier pipeline.</p> <p>Setting up the xavier pipeline is fast and easy! In its most basic form, <code>xavier <b>run</b></code> only has <em>four required inputs</em>.</p> <h2 id=2-synopsis>2. Synopsis<a class=headerlink href=#2-synopsis title="Permanent link">&para;</a></h2> <div class=highlight><pre><span></span><code>$ xavier run [--help] \
                   [--mode {local, slurm}] \
                   [--job-name JOB_NAME] \
                   [--callers {mutect2,mutect,strelka, ...}] \
                   [--pairs PAIRS] \
                   [--ffpe] \
                   [--cnv] \
                   [--silent] \
                   [--singularity-cache SINGULARITY_CACHE] \
                   [--sif-cache SIF_CACHE] \
                   [--threads THREADS] \
                   --runmode {init, dryrun, run} \
                   --input INPUT [INPUT ...] \
                   --output OUTPUT \
                   --genome {hg38, ...} \
                   --targets TARGETS
</code></pre></div> <p>The synopsis for each command shows its parameters and their usage. Optional parameters are shown in square brackets.</p> <p>A user <strong>must</strong> provide a list of FastQ or BAM files (globbing is supported) to analyze via <code>--input</code> argument, an output directory to store results via <code>--output</code> argument, an exome targets BED file for the samples' capture kit, and select reference genome for alignment and annotation via the <code>--genome</code> argument.</p> <p>Use you can always use the <code>-h</code> option for information on a specific command.</p> <h3 id=21-required-arguments>2.1 Required Arguments<a class=headerlink href=#21-required-arguments title="Permanent link">&para;</a></h3> <p>Each of the following arguments are required. Failure to provide a required argument will result in a non-zero exit-code.</p> <p><code>--input INPUT [INPUT ...]</code></p> <blockquote> <p><strong>Input FastQ or BAM file(s) to process.</strong><br> <em>type: file(s)</em></p> <p>One or more FastQ files can be provided. The pipeline does NOT support single-end WES data. Please provide either a set of FastQ files or a set of BAM files. The pipeline does NOT support processing a mixture of FastQ files and BAM files. From the command-line, each input file should separated by a space. Globbing is supported! This makes selecting FastQ files easy. Input FastQ files should be gzipp-ed.</p> <p><strong><em>Example:</em></strong> <code>--input tests/data/*.R?.fastq.gz</code></p> <p><strong><em>Example:</em></strong> <code>--input /data/CCBR_Pipeliner/testdata/XAVIER/human_subset/*.R?.fastq.gz</code></p> </blockquote> <hr> <p><code>--output OUTPUT</code></p> <blockquote> <p><strong>Path to an output directory.</strong><br> <em>type: path</em></p> <p>This location is where the pipeline will create all of its output files, also known as the pipeline's working directory. If the provided output directory does not exist, it will be initialized automatically.</p> <p><strong><em>Example:</em></strong> <code>--output /data/$USER/WES_hg38</code></p> </blockquote> <hr> <p><code>--runmode {init,dryrun,run}</code> `</p> <blockquote> <p><strong>Execution Process.</strong><br> <em>type: string</em></p> <p>User should initialize the pipeline folder by first running <code>--runmode init</code><br> User should then perform a dry-run to list all steps the pipeline will take<code>--runmode dryrun</code><br> User should then perform the full run <code>--runmode run</code></p> <p><strong><em>Example:</em></strong> <code>--runmode init</code> <em>THEN</em> <code>--runmode dryrun</code> <em>THEN</em> <code>--runmode run</code></p> </blockquote> <hr> <p><code>--genome {hg38, custom.json}</code></p> <blockquote> <p><strong>Reference genome.</strong><br> <em>type: string/file</em></p> <p>This option defines the reference genome for your set of samples. On Biowulf, xavier does comes bundled with pre built reference files for human samples; however, it is worth noting that the pipeline does accept a pre-built resource bundle pulled with the cache sub command (coming soon). Currently, the pipeline only supports the human reference hg38; however, support for mouse reference mm10 will be added soon.</p> <p><strong><em>Pre built Option</em></strong><br> Here is a list of available pre built genomes on Biowulf: hg38, mm10.</p> <p><strong><em>Custom Option</em></strong><br> For users running the pipeline outside of Biowulf, a pre-built resource bundle can be pulled with the cache sub command (coming soon). Please supply the custom reference JSON file that was generated by the cache sub command.</p> <p><strong><em>Example:</em></strong> <code>--genome hg38</code> <em>OR</em> <code>--genome /data/${USER}/hg38/hg38.json</code></p> </blockquote> <hr> <p><code>--targets TARGETS</code></p> <blockquote> <p><strong>Exome targets BED file.</strong><br> <em>type: file</em></p> <p>This file can be obtained from the manufacturer of the target capture kit that was used.</p> <p>If not provided, the default targets file from the genome config is used</p> <p><strong><em>Example:</em></strong> <code>--targets resources/Agilent_SSv7_allExons_hg38.bed</code> &gt; <strong><em>Example:</em></strong> <code>--targets resources/SureSelect_mm10_sorted.bed</code></p> </blockquote> <h3 id=22-options>2.2 Options<a class=headerlink href=#22-options title="Permanent link">&para;</a></h3> <p>Each of the following arguments are optional and do not need to be provided.</p> <p><code>-h, --help</code></p> <blockquote> <p><strong>Display Help.</strong><br> <em>type: boolean flag</em></p> <p>Shows command's synopsis, help message, and an example command</p> <p><strong><em>Example:</em></strong> <code>--help</code></p> </blockquote> <hr> <p><code>--silent</code></p> <blockquote> <p><strong>Silence standard output.</strong><br> <em>type: boolean flag</em></p> <p>Reduces the amount of information directed to standard output when submitting master job to the job scheduler. Only the job id of the master job is returned.</p> <p><strong><em>Example:</em></strong> <code>--silent</code></p> </blockquote> <hr> <p><code>--mode {local,slurm}</code></p> <blockquote> <p><strong>Execution Method.</strong><br> <em>type: string</em><br> <em>default: slurm</em></p> <p>Execution Method. Defines the mode or method of execution. Valid mode options include: local or slurm.</p> <p><strong><em>local</em></strong><br> Local executions will run serially on compute instance. This is useful for testing, debugging, or when a users does not have access to a high performance computing environment. If this option is not provided, it will default to a local execution mode.</p> <p><strong><em>slurm</em></strong><br> The slurm execution method will submit jobs to a cluster using a singularity backend. It is recommended running xavier in this mode as execution will be significantly faster in a distributed environment.</p> <p><strong><em>Example:</em></strong> <code>--mode slurm</code></p> </blockquote> <hr> <p><code>--job-name JOB_NAME</code></p> <blockquote> <p><strong>Set the name of the pipeline's master job.</strong><br> <em>type: string</em> &gt; <em>default: pl:xavier</em></p> <p>When submitting the pipeline to a job scheduler, like SLURM, this option always you to set the name of the pipeline's master job. By default, the name of the pipeline's master job is set to "pl:xavier".</p> <p><strong><em>Example:</em></strong> <code>--job-name xavier_run1</code></p> </blockquote> <hr> <p><code>--callers CALLERS [CALLERS ...]</code></p> <blockquote> <p><strong>Variant Callers.</strong><br> <em>type: string(s)</em> &gt; <em>default: mutect2, mutect, strelka, vardict, varscan</em></p> <p>List of variant callers to detect mutations. Please select from one or more of the following options: [mutect2, mutect, strelka, vardict, varscan]. Defaults to using all variant callers.</p> <p><strong><em>Example:</em></strong> <code>--callers mutect2 strelka varscan</code></p> </blockquote> <hr> <p><code>--pairs PAIRS</code></p> <blockquote> <p><strong>Tumor normal pairs file.</strong><br> <em>type: file</em></p> <p>This tab delimited file contains two columns with the names of tumor and normal pairs, one per line. The header of the file needs to be <code>Tumor</code> for the tumor column and <code>Normal</code> for the normal column. The base name of each sample should be listed in the pairs file. The base name of a given sample can be determined by removing the following extension from the sample's R1 FastQ file: <code>.R1.fastq.gz</code>.<br> <strong>Contents of example pairs file:</strong></p> <div class=highlight><pre><span></span><code>Normal    Tumor
Sample4_CRL1622_S31   Sample10_ARK1_S37
Sample4_CRL1622_S31   Sample11_ACI_158_S38
</code></pre></div> <p><strong><em>Example:</em></strong> <code>--pairs /data/$USER/pairs.tsv</code></p> </blockquote> <hr> <p><code>--ffpe</code></p> <blockquote> <p><strong>Apply FFPE correction.</strong><br> <em>type: boolean flag</em></p> <p>Runs an additional steps to correct strand orientation bias in Formalin-Fixed Paraffin-Embedded (FFPE) samples. Do NOT use this option with non-FFPE samples.</p> <p><strong><em>Example:</em></strong> <code>--ffpe</code></p> </blockquote> <hr> <p><code>--cnv</code></p> <blockquote> <p><strong>Call copy number variations (CNVs).</strong><br> <em>type: boolean flag</em></p> <p>CNVs will only be called from tumor-normal pairs. If this option is provided without providing a --pairs file, CNVs will NOT be called.</p> <p><strong><em>Example:</em></strong> <code>--cnv</code></p> </blockquote> <hr> <p><code>--singularity-cache SINGULARITY_CACHE</code></p> <blockquote> <p><strong>Overrides the $SINGULARITY_CACHEDIR environment variable.</strong><br> <em>type: path</em><br> <em>default: <code>--output OUTPUT/.singularity</code></em></p> <p>Singularity will cache image layers pulled from remote registries. This ultimately speeds up the process of pull an image from DockerHub if an image layer already exists in the singularity cache directory. By default, the cache is set to the value provided to the <code>--output</code> argument. Please note that this cache cannot be shared across users. Singularity strictly enforces you own the cache directory and will return a non-zero exit code if you do not own the cache directory! See the <code>--sif-cache</code> option to create a shareable resource.</p> <p><strong><em>Example:</em></strong> <code>--singularity-cache /data/$USER/.singularity</code></p> </blockquote> <hr> <p><code>--sif-cache SIF_CACHE</code></p> <blockquote> <p><strong>Path where a local cache of SIFs are stored.</strong><br> <em>type: path</em></p> <p>Uses a local cache of SIFs on the filesystem. This SIF cache can be shared across users if permissions are set correctly. If a SIF does not exist in the SIF cache, the image will be pulled from Dockerhub and a warning message will be displayed. The <code>xavier cache</code> subcommand can be used to create a local SIF cache. Please see <code>xavier cache</code> for more information. This command is extremely useful for avoiding DockerHub pull rate limits. It also remove any potential errors that could occur due to network issues or DockerHub being temporarily unavailable. We recommend running xavier with this option when ever possible.</p> <p><strong><em>Example:</em></strong> <code>--singularity-cache /data/$USER/SIFs</code></p> </blockquote> <hr> <p><code>--threads THREADS</code></p> <blockquote> <p><strong>Max number of threads for each process.</strong><br> <em>type: int</em><br> <em>default: 2</em></p> <p>Max number of threads for each process. This option is more applicable when running the pipeline with <code>--mode local</code>. It is recommended setting this value to the maximum number of CPUs available on the host machine.</p> <p><strong><em>Example:</em></strong> <code>--threads 12</code></p> </blockquote> <h2 id=3-example>3. Example<a class=headerlink href=#3-example title="Permanent link">&para;</a></h2> <div class=highlight><pre><span></span><code><span class=c1># Step 1.) Grab an interactive node</span>
<span class=c1># Do not run on head node!</span>
sinteractive<span class=w> </span>--mem<span class=o>=</span>8g<span class=w> </span>--cpus-per-task<span class=o>=</span><span class=m>4</span>
module<span class=w> </span>purge
module<span class=w> </span>load<span class=w> </span>ccbrpipeliner

<span class=c1># Step 2A.) Initialize the all resources to the output folder</span>
xavier<span class=w> </span>run<span class=w> </span>--input<span class=w> </span>tests/data/*.R?.fastq.gz<span class=w> </span><span class=se>\</span>
<span class=w>                 </span>--output<span class=w> </span>/data/<span class=nv>$USER</span>/xavier_hg38<span class=w> </span><span class=se>\</span>
<span class=w>                 </span>--genome<span class=w> </span>hg38<span class=w> </span><span class=se>\</span>
<span class=w>                 </span>--targets<span class=w> </span>Agilent_SSv7_allExons_hg38.bed<span class=w> </span><span class=se>\</span>
<span class=w>                 </span>--mode<span class=w> </span>slurm<span class=w> </span><span class=se>\</span>
<span class=w>                 </span>--runmode<span class=w> </span>init

<span class=c1># Step 2B.) Dry-run the pipeline</span>
xavier<span class=w> </span>run<span class=w> </span>--input<span class=w> </span>tests/data/*.R?.fastq.gz<span class=w> </span><span class=se>\</span>
<span class=w>                 </span>--output<span class=w> </span>/data/<span class=nv>$USER</span>/xavier_hg38<span class=w> </span><span class=se>\</span>
<span class=w>                 </span>--genome<span class=w> </span>hg38<span class=w> </span><span class=se>\</span>
<span class=w>                 </span>--targets<span class=w> </span>Agilent_SSv7_allExons_hg38.bed<span class=w> </span><span class=se>\</span>
<span class=w>                 </span>--mode<span class=w> </span>slurm<span class=w> </span><span class=se>\</span>
<span class=w>                 </span>--runmode<span class=w> </span>dryrun

<span class=c1># Step 2C.) Run the XAVIER pipeline</span>
<span class=c1># The slurm mode will submit jobs to the cluster.</span>
<span class=c1># It is recommended running xavier in this mode.</span>
xavier<span class=w> </span>run<span class=w> </span>--input<span class=w> </span>tests/data/*.R?.fastq.gz<span class=w> </span><span class=se>\</span>
<span class=w>                 </span>--output<span class=w> </span>/data/<span class=nv>$USER</span>/xavier_hg38<span class=w> </span><span class=se>\</span>
<span class=w>                 </span>--genome<span class=w> </span>hg38<span class=w> </span><span class=se>\</span>
<span class=w>                 </span>--targets<span class=w> </span>Agilent_SSv7_allExons_hg38.bed<span class=w> </span><span class=se>\</span>
<span class=w>                 </span>--mode<span class=w> </span>slurm<span class=w> </span><span class=se>\</span>
<span class=w>                 </span>--runmode<span class=w> </span>run
</code></pre></div> <p>The example dataset in <code>tests/data</code> in this repository is a very small subsampled dataset, and some steps of the pipeline fail due to the small size (CNV callling, somalier, etc). We have a larger subsample (25% of a full human dataset) available on Biowulf if you would like to test the full functionality of the pipeline: <code>/data/CCBR_Pipeliner/testdata/XAVIER/human_subset/*.R?.fastq.gz</code></p> <aside class=md-source-file> <span class=md-source-file__fact> <span class=md-icon title="Last update"> <svg xmlns=http://www.w3.org/2000/svg viewbox="0 0 24 24"><path d="M21 13.1c-.1 0-.3.1-.4.2l-1 1 2.1 2.1 1-1c.2-.2.2-.6 0-.8l-1.3-1.3c-.1-.1-.2-.2-.4-.2m-1.9 1.8-6.1 6V23h2.1l6.1-6.1-2.1-2M12.5 7v5.2l4 2.4-1 1L11 13V7h1.5M11 21.9c-5.1-.5-9-4.8-9-9.9C2 6.5 6.5 2 12 2c5.3 0 9.6 4.1 10 9.3-.3-.1-.6-.2-1-.2s-.7.1-1 .2C19.6 7.2 16.2 4 12 4c-4.4 0-8 3.6-8 8 0 4.1 3.1 7.5 7.1 7.9l-.1.2v1.8Z"/></svg> </span> 2024-08-16 </span> </aside> </article> </div> <script>var target=document.getElementById(location.hash.slice(1));target&&target.name&&(target.checked=target.name.startsWith("__tabbed_"))</script> </div> <button type=button class="md-top md-icon" data-md-component=top hidden> <svg xmlns=http://www.w3.org/2000/svg viewbox="0 0 24 24"><path d="M13 20h-2V8l-5.5 5.5-1.42-1.42L12 4.16l7.92 7.92-1.42 1.42L13 8v12Z"/></svg> Back to top </button> </main> <footer class=md-footer> <div class="md-footer-meta md-typeset"> <div class="md-footer-meta__inner md-grid"> <div class=md-copyright> <div class=md-copyright__highlight> Copyright &copy; 2023 CCBR </div> Made with <a href=https://squidfunk.github.io/mkdocs-material/ target=_blank rel=noopener> Material for MkDocs </a> </div> <div class=md-social> <a href=http://bioinformatics.cancer.gov target=_blank rel=noopener title=bioinformatics.cancer.gov class=md-social__link> <svg xmlns=http://www.w3.org/2000/svg viewbox="0 0 640 512"><!-- Font Awesome Free 6.5.1 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2023 Fonticons, Inc.--><path d="M144 0a80 80 0 1 1 0 160 80 80 0 1 1 0-160zm368 0a80 80 0 1 1 0 160 80 80 0 1 1 0-160zM0 298.7C0 239.8 47.8 192 106.7 192h42.7c15.9 0 31 3.5 44.6 9.7-1.3 7.2-1.9 14.7-1.9 22.3 0 38.2 16.8 72.5 43.3 96H21.3C9.6 320 0 310.4 0 298.7zM405.3 320h-.7c26.6-23.5 43.3-57.8 43.3-96 0-7.6-.7-15-1.9-22.3 13.6-6.3 28.7-9.7 44.6-9.7h42.7c58.9 0 106.7 47.8 106.7 106.7 0 11.8-9.6 21.3-21.3 21.3H405.3zM224 224a96 96 0 1 1 192 0 96 96 0 1 1-192 0zm-96 261.3c0-73.6 59.7-133.3 133.3-133.3h117.4c73.6 0 133.3 59.7 133.3 133.3 0 14.7-11.9 26.7-26.7 26.7H154.7c-14.7 0-26.7-11.9-26.7-26.7z"/></svg> </a> <a href=https://github.com/CCBR target=_blank rel=noopener title=github.com class=md-social__link> <svg xmlns=http://www.w3.org/2000/svg viewbox="0 0 496 512"><!-- Font Awesome Free 6.5.1 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2023 Fonticons, Inc.--><path d="M165.9 397.4c0 2-2.3 3.6-5.2 3.6-3.3.3-5.6-1.3-5.6-3.6 0-2 2.3-3.6 5.2-3.6 3-.3 5.6 1.3 5.6 3.6zm-31.1-4.5c-.7 2 1.3 4.3 4.3 4.9 2.6 1 5.6 0 6.2-2s-1.3-4.3-4.3-5.2c-2.6-.7-5.5.3-6.2 2.3zm44.2-1.7c-2.9.7-4.9 2.6-4.6 4.9.3 2 2.9 3.3 5.9 2.6 2.9-.7 4.9-2.6 4.6-4.6-.3-1.9-3-3.2-5.9-2.9zM244.8 8C106.1 8 0 113.3 0 252c0 110.9 69.8 205.8 169.5 239.2 12.8 2.3 17.3-5.6 17.3-12.1 0-6.2-.3-40.4-.3-61.4 0 0-70 15-84.7-29.8 0 0-11.4-29.1-27.8-36.6 0 0-22.9-15.7 1.6-15.4 0 0 24.9 2 38.6 25.8 21.9 38.6 58.6 27.5 72.9 20.9 2.3-16 8.8-27.1 16-33.7-55.9-6.2-112.3-14.3-112.3-110.5 0-27.5 7.6-41.3 23.6-58.9-2.6-6.5-11.1-33.3 2.6-67.9 20.9-6.5 69 27 69 27 20-5.6 41.5-8.5 62.8-8.5s42.8 2.9 62.8 8.5c0 0 48.1-33.6 69-27 13.7 34.7 5.2 61.4 2.6 67.9 16 17.7 25.8 31.5 25.8 58.9 0 96.5-58.9 104.2-114.8 110.5 9.2 7.9 17 22.9 17 46.4 0 33.7-.3 75.4-.3 83.6 0 6.5 4.6 14.4 17.3 12.1C428.2 457.8 496 362.9 496 252 496 113.3 383.5 8 244.8 8zM97.2 352.9c-1.3 1-1 3.3.7 5.2 1.6 1.6 3.9 2.3 5.2 1 1.3-1 1-3.3-.7-5.2-1.6-1.6-3.9-2.3-5.2-1zm-10.8-8.1c-.7 1.3.3 2.9 2.3 3.9 1.6 1 3.6.7 4.3-.7.7-1.3-.3-2.9-2.3-3.9-2-.6-3.6-.3-4.3.7zm32.4 35.6c-1.6 1.3-1 4.3 1.3 6.2 2.3 2.3 5.2 2.6 6.5 1 1.3-1.3.7-4.3-1.3-6.2-2.2-2.3-5.2-2.6-6.5-1zm-11.4-14.7c-1.6 1-1.6 3.6 0 5.9 1.6 2.3 4.3 3.3 5.6 2.3 1.6-1.3 1.6-3.9 0-6.2-1.4-2.3-4-3.3-5.6-2z"/></svg> </a> <a href=https://hub.docker.com/orgs/nciccbr/repositories target=_blank rel=noopener title=hub.docker.com class=md-social__link> <svg xmlns=http://www.w3.org/2000/svg viewbox="0 0 640 512"><!-- Font Awesome Free 6.5.1 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2023 Fonticons, Inc.--><path d="M349.9 236.3h-66.1v-59.4h66.1v59.4zm0-204.3h-66.1v60.7h66.1V32zm78.2 144.8H362v59.4h66.1v-59.4zm-156.3-72.1h-66.1v60.1h66.1v-60.1zm78.1 0h-66.1v60.1h66.1v-60.1zm276.8 100c-14.4-9.7-47.6-13.2-73.1-8.4-3.3-24-16.7-44.9-41.1-63.7l-14-9.3-9.3 14c-18.4 27.8-23.4 73.6-3.7 103.8-8.7 4.7-25.8 11.1-48.4 10.7H2.4c-8.7 50.8 5.8 116.8 44 162.1 37.1 43.9 92.7 66.2 165.4 66.2 157.4 0 273.9-72.5 328.4-204.2 21.4.4 67.6.1 91.3-45.2 1.5-2.5 6.6-13.2 8.5-17.1l-13.3-8.9zm-511.1-27.9h-66v59.4h66.1v-59.4zm78.1 0h-66.1v59.4h66.1v-59.4zm78.1 0h-66.1v59.4h66.1v-59.4zm-78.1-72.1h-66.1v60.1h66.1v-60.1z"/></svg> </a> </div> </div> </div> </footer> </div> <div class=md-dialog data-md-component=dialog> <div class="md-dialog__inner md-typeset"></div> </div> <script id=__config type=application/json>{"base": "../..", "features": ["navigation.tabs", "navigation.top", "toc.integrate"], "search": "../../assets/javascripts/workers/search.b8dbb3d2.min.js", "translations": {"clipboard.copied": "Copied to clipboard", "clipboard.copy": "Copy to clipboard", "search.result.more.one": "1 more on this page", "search.result.more.other": "# more on this page", "search.result.none": "No matching documents", "search.result.one": "1 matching document", "search.result.other": "# matching documents", "search.result.placeholder": "Type to start searching", "search.result.term.missing": "Missing", "select.version": "Select version"}, "version": {"provider": "mike"}}</script> <script src=../../assets/javascripts/bundle.e1c3ead8.min.js></script> </body> </html>