<!DOCTYPE html><html class=writer-html5 lang=en> <head><meta charset=utf-8><meta http-equiv=X-UA-Compatible content="IE=edge"><meta name=viewport content="width=device-width, initial-scale=1.0"><meta name=author content=CCBR><link rel="shortcut icon" href=../../img/favicon.ico><title>Output Files - XAVIER Documentation</title><link rel=stylesheet href=../../css/theme.css><link rel=stylesheet href=../../css/theme_extra.css><link rel=stylesheet href=https://cdnjs.cloudflare.com/ajax/libs/highlight.js/11.8.0/styles/github.min.css><link href=../../css/custom.css rel=stylesheet><link href=../../css/version-select.css rel=stylesheet><script>
        // Current page data
        var mkdocs_page_name = "Output Files";
        var mkdocs_page_input_path = "pipeline-details/output.md";
        var mkdocs_page_url = null;
      </script><!--[if lt IE 9]>
      <script src="../../js/html5shiv.min.js"></script>
    <![endif]--><script src=https://cdnjs.cloudflare.com/ajax/libs/highlight.js/11.8.0/highlight.min.js></script><script>hljs.highlightAll();</script></head> <body class=wy-body-for-nav role=document> <div class=wy-grid-for-nav> <nav data-toggle=wy-nav-shift class="wy-nav-side stickynav"> <div class=wy-side-scroll> <div class=wy-side-nav-search> <a href=../.. class="icon icon-home"> XAVIER Documentation </a><div role=search> <form id=rtd-search-form class=wy-form action=../../search.html method=get> <input type=text name=q placeholder="Search docs" aria-label="Search docs" title="Type search term here"> </form> </div> </div> <div class="wy-menu wy-menu-vertical" data-spy=affix role=navigation aria-label="Navigation menu"> <ul> <li class=toctree-l1><a class="reference internal" href=../..>About</a> </li> </ul> <p class=caption><span class=caption-text>Command Line</span></p> <ul> <li class=toctree-l1><a href=../../usage/run/ class="reference internal">xavier run</a> </li> <li class=toctree-l1><a href=../../usage/unlock/ class="reference internal">xavier unlock</a> </li> <li class=toctree-l1><a href=../../usage/cache/ class="reference internal">xavier cache</a> </li> </ul> <ul> <li class=toctree-l1><a href=../../usage/gui/ class="reference internal">Graphical Interface</a> </li> </ul> <p class=caption><span class=caption-text>Pipeline Details</span></p> <ul class=current> <li class=toctree-l1><a href=../overview/ class="reference internal">Overview</a> </li> <li class=toctree-l1><a href=../methods/ class="reference internal">Methods</a> </li> <li class=toctree-l1><a href=../tools/ class="reference internal">Tools</a> </li> <li class=toctree-l1><a href=../settings/ class="reference internal">Settings</a> </li> <li class="toctree-l1 current"><a class="reference internal current" href=#>Output Files</a> <ul class=current> <li class=toctree-l2><a class="reference internal" href=#xavier>XAVIER</a> <ul> <li class=toctree-l3><a class="reference internal" href=#1-qc>1. QC</a> </li> <li class=toctree-l3><a class="reference internal" href=#2-bams>2. bams</a> </li> <li class=toctree-l3><a class="reference internal" href=#3-germline>3. germline</a> </li> <li class=toctree-l3><a class="reference internal" href=#4-logfiles>4. logfiles</a> </li> </ul> </li> <li class=toctree-l2><a class="reference internal" href=#tumor-normal-pair>Tumor-normal pair</a> <ul> <li class=toctree-l3><a class="reference internal" href=#somatic_paired>somatic_paired</a> </li> </ul> </li> <li class=toctree-l2><a class="reference internal" href=#tumor-only>Tumor-only</a> <ul> <li class=toctree-l3><a class="reference internal" href=#somatic_tumor_only>somatic_tumor_only</a> </li> </ul> </li> </ul> </li> </ul> <p class=caption><span class=caption-text>FAQ</span></p> <ul> <li class=toctree-l1><a href=../../contributing/ class="reference internal">How to contribute</a> </li> </ul> <ul> <li class=toctree-l1><a href=../../license/ class="reference internal">License</a> </li> </ul> </div> </div> </nav> <section data-toggle=wy-nav-shift class=wy-nav-content-wrap> <nav class=wy-nav-top role=navigation aria-label="Mobile navigation menu"> <i data-toggle=wy-nav-top class="fa fa-bars"></i> <a href=../..>XAVIER Documentation</a> </nav> <div class=wy-nav-content> <div class=rst-content><div role=navigation aria-label="breadcrumbs navigation"> <ul class=wy-breadcrumbs> <li><a href=../.. class="icon icon-home" aria-label=Docs></a></li> <li class=breadcrumb-item>Pipeline Details</li> <li class="breadcrumb-item active">Output Files</li> <li class=wy-breadcrumbs-aside> <a href=https://github.com/CCBR/XAVIER/edit/unified/docs/pipeline-details/output.md>Edit on CCBR/XAVIER</a> </li> </ul> <hr> </div> <div role=main class=document itemscope=itemscope itemtype=http://schema.org/Article> <div class=section itemprop=articleBody> <h1 id=output-files>Output files<a class=headerlink href=#output-files title="Permanent link">&para;</a></h1> <h2 id=xavier>XAVIER<a class=headerlink href=#xavier title="Permanent link">&para;</a></h2> <p>The output files and their locations are broken down here for the XAVIER pipeline. Pre-processing and germline variant calling steps are common but somatic variant calling is dependent on whether the pipeline was run in either (A) tumor-normal pair or (B) tumor-only analysis mode. All file locations are relative to the output directory specified during the job submission.</p> <p>The output directory after a complete XAVIER run should look like:</p> <div class=highlight><pre><span></span><code>xavier_output/
├──<span class=w> </span>bams
├──<span class=w> </span>cluster.json<span class=w> </span><span class=c1># cluster info for the run</span>
├──<span class=w> </span>config
├──<span class=w> </span>config.json<span class=w>  </span><span class=c1># config file for the run</span>
├──<span class=w> </span>fastqs
├──<span class=w> </span>germline
├──<span class=w> </span>indels.vcf.gz<span class=o>[</span>.tbi<span class=o>]</span><span class=w> </span><span class=c1># raw germline INDELs</span>
├──<span class=w> </span>input_files
├──<span class=w> </span>intervals.list
├──<span class=w> </span><span class=o>{</span>sample1<span class=o>}</span>-normal.R1.fastq.gz<span class=w> </span>-&gt;<span class=w> </span>/path/to/<span class=o>{</span>sample1<span class=o>}</span>-normal.R1.fastq.gz
├──<span class=w> </span><span class=o>{</span>sample1<span class=o>}</span>-normal.R2.fastq.gz<span class=w> </span>-&gt;<span class=w> </span>/path/to/<span class=o>{</span>sample1<span class=o>}</span>-normal.R2.fastq.gz
├──<span class=w> </span><span class=o>{</span>sample1<span class=o>}</span>-tumor.R1.fastq.gz<span class=w> </span>-&gt;<span class=w> </span>/path/to/<span class=o>{</span>sample1<span class=o>}</span>-tumor.R1.fastq.gz
├──<span class=w> </span><span class=o>{</span>sample1<span class=o>}</span>-tumor.R2.fastq.gz<span class=w> </span>-&gt;<span class=w> </span>/path/to/<span class=o>{</span>sample1<span class=o>}</span>-tumor.R2.fastq.gz
.
.
.
├──<span class=w> </span>kickoff.sh
├──<span class=w> </span>logfiles
├──<span class=w> </span>QC
├──<span class=w> </span>resources
├──<span class=w> </span>snps.vcf.gz<span class=o>[</span>.tbi<span class=o>]</span><span class=w> </span><span class=c1># raw germline SNPs</span>
├──<span class=w> </span>somatic_paired<span class=w> </span><span class=c1># in case of tumor-normal paired run</span>
├──<span class=w> </span>somatic_tumor_only<span class=w> </span><span class=c1># in case of tumor-only run</span>
└──<span class=w> </span>workflow
</code></pre></div> <p>Below we describe the different folders that contain specific outputs obtained for all samples from the XAVIER pipeline</p> <h3 id=1-qc>1. <code>QC</code><a class=headerlink href=#1-qc title="Permanent link">&para;</a></h3> <p>The <code>QC</code> folder contains all the Quality-Control analyses performed at different steps of the pipeline for each sample to assess sequencing quality before and after adapter trimming, microbial taxonomic composition, contamination, variant calling, etc. The final summary report and data is available <code>finalQC</code> folder. \ The MultiQC report also contains results from other analysis like mapping statistics, ancestry and relatedness, etc. It is recommended to study the MultiQC report first to get a birds eye view of the sequence data quality.</p> <div class=highlight><pre><span></span><code>QC/
├──<span class=w> </span>exome_targets.bed
├──<span class=w> </span>finalQC/
│<span class=w>   </span>├──<span class=w> </span>MultiQC_Report_data
│<span class=w>   </span>└──<span class=w> </span>MultiQC_Report.html
├──<span class=w> </span>FQscreen
├──<span class=w> </span>sample1-normal
├──<span class=w> </span>sample1-normal_fastqc.html
.
.
.
├──<span class=w> </span>sample1-tumor
├──<span class=w> </span>sample1-tumor_fastqc.html
.
.
.
├──<span class=w> </span>kraken
├──<span class=w> </span>raw_variants.het
├──<span class=w> </span>raw_variants.variant_calling_detail_metrics
└──<span class=w> </span>raw_variants.variant_calling_summary_metrics
</code></pre></div> <h3 id=2-bams>2. <code>bams</code><a class=headerlink href=#2-bams title="Permanent link">&para;</a></h3> <p>The <code>bams</code> folder contain two subfolders <code>chrom_split</code> and <code>final_bams</code>. <code>final_bams</code> contains the final processed BAM files for each sample in the run and the <code>chrom_split</code> folder contains all the sample BAM files split by each chromosome.</p> <div class=highlight><pre><span></span><code>bams/
├──<span class=w> </span>chrom_split
│<span class=w>   </span>├──<span class=w> </span><span class=o>{</span>sample1<span class=o>}</span>-normal.chr1.split.bam<span class=o>[</span>.bai<span class=o>]</span>
│<span class=w>   </span>├──<span class=w> </span><span class=o>{</span>sample1<span class=o>}</span>-normal.chr2.split.bam<span class=o>[</span>.bai<span class=o>]</span>
.<span class=w>   </span>.
.<span class=w>   </span>.
.<span class=w>   </span>.
│<span class=w>   </span>└──<span class=w> </span><span class=o>{</span>sampleN<span class=o>}</span>-tumor.chrN.split.bam<span class=o>[</span>.bai<span class=o>]</span>
└──<span class=w> </span>final_bams
<span class=w>    </span>├──<span class=w> </span><span class=o>{</span>sample1<span class=o>}</span>-normal.bam<span class=o>[</span>.bai<span class=o>]</span>
<span class=w>    </span>├──<span class=w> </span><span class=o>{</span>sample1<span class=o>}</span>-tumor.bam<span class=o>[</span>.bai<span class=o>]</span>
<span class=w>    </span>.
<span class=w>    </span>.
<span class=w>    </span>.
<span class=w>    </span>└──<span class=w> </span><span class=o>{</span>sampleN<span class=o>}</span>-tumor.bam.bai
</code></pre></div> <h3 id=3-germline>3. <code>germline</code><a class=headerlink href=#3-germline title="Permanent link">&para;</a></h3> <p>This folder contains the output from the GATK Best Practices pipeline to obtain germline variants with a few alterations detailed below. Briefly, joint SNP and INDEL variant detection is conducted across all samples included in a pipeline run using the GATK Haplotypcaller under default settings. Raw variants are then subsequently filtered based on several GATK annotations: \ A strict set of criteria (QD &lt; 2.0, FS &gt; 60.0, MQ &lt; 40.0, MQRankSum &lt; -12.5, ReadPosRankSum &lt; -8.0 for SNPs; QD &lt; 2.0, FS &gt; 200.0, ReadPosRankSum &lt; -20.0 for INDELs) generates the 'combined.strictFilter.vcf'.</p> <p>This call set is highly stringent, maximizing the true positive rate at the expense of an elevated false negative rate. This call set is really only intended for more general population genetic scale analyses (e.g., burden tests, admixture, linkage/pedigree based analysis, etc.) where false positives can be significantly confounding.</p> <p>In case of human sequence data, a basic analyses of sample relatedness and ancestry (e.g., % European, African, etc.) is also performed using somalier.</p> <p>The output folder looks like:</p> <div class=highlight><pre><span></span><code>germline/
├──<span class=w> </span>gVCFs
.
.
.
├──<span class=w> </span>somalier<span class=w> </span><span class=c1># only for hg38 genome</span>
└──<span class=w> </span>VCF
</code></pre></div> <p>The <code>VCF</code> folder contains the final filtered germline variants (SNPs and INDELs) for all samples combined. The folder also contains raw variants for each sample, all samples combined, and also combined raw variants split by chromosome.</p> <div class=highlight><pre><span></span><code>VCF/
├──<span class=w> </span>by_chrom
│<span class=w>   </span>├──<span class=w> </span>raw_variants_byChrom.list
│<span class=w>   </span>├──<span class=w> </span>raw_variants.chr1.vcf.gz<span class=o>[</span>.tbi<span class=o>]</span>
.<span class=w>   </span>.
.<span class=w>   </span>.
.<span class=w>   </span>.
│<span class=w>   </span>└──<span class=w> </span>raw_variants.chrN.vcf.gz<span class=o>[</span>.tbi<span class=o>]</span>
├──<span class=w> </span>indel.filterd.vcf.gz<span class=o>[</span>.tbi<span class=o>]</span>
├──<span class=w> </span><span class=o>{</span>sample1<span class=o>}</span>-normal.germline.vcf.gz<span class=o>[</span>.tbi<span class=o>]</span>
├──<span class=w> </span><span class=o>{</span>sample1<span class=o>}</span>-tumor.germline.vcf.gz<span class=o>[</span>.tbi<span class=o>]</span>
.
.
.
├──<span class=w> </span><span class=o>{</span>sampleN<span class=o>}</span>-normal.germline.vcf.gz<span class=o>[</span>.tbi<span class=o>]</span>
├──<span class=w> </span><span class=o>{</span>sampleN<span class=o>}</span>-tumor.germline.vcf.gz<span class=o>[</span>.tbi<span class=o>]</span>
├──<span class=w> </span>raw_variants.vcf.gz<span class=o>[</span>.tbi<span class=o>]</span>
├──<span class=w> </span>snp.filtered.vcf.gz<span class=o>[</span>.tbi<span class=o>]</span>
└──<span class=w> </span>snp_indel.filtered.vcf.gz<span class=o>[</span>.tbi<span class=o>]</span>
</code></pre></div> <h3 id=4-logfiles>4. <code>logfiles</code><a class=headerlink href=#4-logfiles title="Permanent link">&para;</a></h3> <p>This folder contains the snakemake log files and computational statistics for the XAVIER run. All the log files (i.e., standard output and error) for each individual step are in the <code>slurmfiles</code> folder. These logfiles are important to diagnose errors in case the pipeline fails.</p> <div class=highlight><pre><span></span><code>logfiles/
├──<span class=w> </span>master.log
├──<span class=w> </span>mjobid.log
├──<span class=w> </span>runtime_statistics.json
├──<span class=w> </span>slurmfiles
├──<span class=w> </span>snakemake.log
├──<span class=w> </span>snakemake.log.jobby
└──<span class=w> </span>snakemake.log.jobby.short
</code></pre></div> <h2 id=tumor-normal-pair>Tumor-normal pair<a class=headerlink href=#tumor-normal-pair title="Permanent link">&para;</a></h2> <h3 id=somatic_paired><code>somatic_paired</code><a class=headerlink href=#somatic_paired title="Permanent link">&para;</a></h3> <p>This workflow calls somatic SNPs and INDELs using multiple variant detection algorithms. For each of these tools, variants are called in a paired tumor-normal fashion, with default settings. See <strong>Pipeline Details</strong> for more information about the tools used and their parameter settings.</p> <p>For each sample, the resulting VCF is fully annotated using VEP and converted to a MAF file using the vcf2maf tool. Resulting MAF files are found in <code>maf</code> folder within each caller's results directory (i.e., <code>mutect2_out</code>, <code>strelka_out</code>, etc.). Individual sample MAF files are then merged and saved in <code>merged_somatic_variants</code> directory.</p> <p>For Mutect2, we use a panel of normals (PON) developed from the ExAC (excluding TCGA) dataset, filtered for variants &lt;0.001 in the general population, and also including and in-house set of blacklisted recurrent germline variants that are not found in any population databases.</p> <p>For Copy Number Variants (CNVs), two tools are employed in tandem. First, Control-FREEC is run with default parameters. This generates pileup files that can be used by Sequenza, primarily for jointly estimating contamination and ploidy. These value are used to run Freec a second time for improved performance.</p> <p>The output directory should look like:</p> <div class=highlight><pre><span></span><code>somatic_paired/
├──<span class=w> </span>CNV<span class=w> </span><span class=c1># only if CNVs analyzed</span>
│<span class=w>   </span>├──<span class=w> </span>freec_out
│<span class=w>   </span>└──<span class=w> </span>sequenza_out
├──<span class=w> </span>ffpe_filter<span class=w> </span><span class=c1># only if FFPE filter applied</span>
├──<span class=w> </span>qc
└──<span class=w> </span>SNP_Indels
<span class=w>    </span>├──<span class=w> </span>merged_somatic_variants
<span class=w>    </span>│<span class=w>   </span>├──<span class=w> </span>cohort_summary
<span class=w>    </span>│<span class=w>   </span>├──<span class=w> </span>maf<span class=w> </span><span class=c1># Final merged MAFs for each sample</span>
<span class=w>    </span>│<span class=w>   </span>└──<span class=w> </span>vcf
<span class=w>    </span>├──<span class=w> </span>mutect2_out
<span class=w>    </span>│<span class=w>   </span>├──<span class=w> </span>chrom_split
<span class=w>    </span>│<span class=w>   </span>├──<span class=w> </span>cohort_summary
<span class=w>    </span>│<span class=w>   </span>├──<span class=w> </span>maf
<span class=w>    </span>│<span class=w>   </span>├──<span class=w> </span>pileup_summaries
<span class=w>    </span>│<span class=w>   </span>├──<span class=w> </span>read_orientation_data
<span class=w>    </span>│<span class=w>   </span>└──<span class=w> </span>vcf
<span class=w>    </span>├──<span class=w> </span>mutect_out
<span class=w>    </span>│<span class=w>   </span>├──<span class=w> </span>maf
<span class=w>    </span>│<span class=w>   </span>├──<span class=w> </span>.
<span class=w>    </span>│<span class=w>   </span>├──<span class=w> </span>.
<span class=w>    </span>│<span class=w>   </span>└──<span class=w> </span>vcf
<span class=w>    </span>├──<span class=w> </span>strelka_out
<span class=w>    </span>├──<span class=w> </span>vardict_out
<span class=w>    </span>└──<span class=w> </span>varscan_out
</code></pre></div> <h2 id=tumor-only>Tumor-only<a class=headerlink href=#tumor-only title="Permanent link">&para;</a></h2> <h3 id=somatic_tumor_only><code>somatic_tumor_only</code><a class=headerlink href=#somatic_tumor_only title="Permanent link">&para;</a></h3> <p>In general, the tumor-only pipeline is a stripped down version of the tumor-normal pipeline. We only run MuTect2, Mutect, and VarDict for somatic variant detection, with the same PON and filtering as described above for the tumor-normal pipeline.</p> <div class=highlight><pre><span></span><code>somatic_tumor_only/
├──<span class=w> </span>ffpe_filter<span class=w> </span><span class=c1># only if FFPE filter applied</span>
├──<span class=w> </span>qc
└──<span class=w> </span>SNP_Indels
<span class=w>    </span>├──<span class=w> </span>merged_somatic_variants
<span class=w>    </span>│<span class=w>   </span>├──<span class=w> </span>cohort_summary
<span class=w>    </span>│<span class=w>   </span>├──<span class=w> </span>maf<span class=w> </span><span class=c1># Final merged MAFs for each sample</span>
<span class=w>    </span>│<span class=w>   </span>└──<span class=w> </span>vcf
<span class=w>    </span>├──<span class=w> </span>mutect2_out
<span class=w>    </span>│<span class=w>   </span>├──<span class=w> </span>chrom_split
<span class=w>    </span>│<span class=w>   </span>├──<span class=w> </span>cohort_summary
<span class=w>    </span>│<span class=w>   </span>├──<span class=w> </span>maf
<span class=w>    </span>│<span class=w>   </span>├──<span class=w> </span>read_orientation_data
<span class=w>    </span>│<span class=w>   </span>└──<span class=w> </span>vcf
<span class=w>    </span>├──<span class=w> </span>mutect_out
<span class=w>    </span>│<span class=w>   </span>├──<span class=w> </span>maf
<span class=w>    </span>│<span class=w>   </span>├──<span class=w> </span>.
<span class=w>    </span>│<span class=w>   </span>├──<span class=w> </span>.
<span class=w>    </span>│<span class=w>   </span>└──<span class=w> </span>vcf
<span class=w>    </span>├──<span class=w> </span>vardict_out
<span class=w>    </span>└──<span class=w> </span>varscan_out
</code></pre></div> </div> </div><script async src="https://www.googletagmanager.com/gtag/js?id=G-D3SL9V30KL"></script> <script>
  window.dataLayer = window.dataLayer || [];
  function gtag(){dataLayer.push(arguments);}
  gtag('js', new Date());
  gtag('config', 'G-D3SL9V30KL'); // Replace with your tracking ID
</script> </div> </div> </section> </div> <div class=rst-versions role=note aria-label=Versions> <span class=rst-current-version data-toggle=rst-current-version> <span> <a href=https://github.com/CCBR/XAVIER class="fa fa-code-fork" style="color: #fcfcfc"> CCBR/XAVIER</a> </span> <span><a href=../settings/ style="color: #fcfcfc">&laquo; Previous</a></span> <span><a href=../../contributing/ style="color: #fcfcfc">Next &raquo;</a></span> </span> </div> <script src=../../js/jquery-3.6.0.min.js></script> <script>var base_url = "../..";</script> <script src=../../js/theme_extra.js></script> <script src=../../js/theme.js></script> <script src=../../js/custom.js></script> <script src=../../search/main.js></script> <script src=../../js/version-select.js></script> <script>
        jQuery(function () {
            SphinxRtdTheme.Navigation.enable(true);
        });
    </script> </body> </html> 