<!doctype html><html lang=en class=no-js> <head><meta charset=utf-8><meta name=viewport content="width=device-width,initial-scale=1"><meta name=description content="XAVIER is an open-source, reproducible, and scalable best practices pipeline for analyzing Whole Exome DNA-sequencing data."><meta name=author content=CCBR><link href=../settings/ rel=prev><link href=../../faq/questions/ rel=next><link rel=icon href=../../assets/favicon/favicon.ico><meta name=generator content="mkdocs-1.5.3, mkdocs-material-9.5.6"><title>Output Files - XAVIER Documentation</title><link rel=stylesheet href=../../assets/stylesheets/main.50c56a3b.min.css><link rel=stylesheet href=../../assets/stylesheets/palette.06af60db.min.css><link rel=preconnect href=https://fonts.gstatic.com crossorigin><link rel=stylesheet href="https://fonts.googleapis.com/css?family=Roboto:300,300i,400,400i,700,700i%7CRoboto+Mono:400,400i,700,700i&display=fallback"><style>:root{--md-text-font:"Roboto";--md-code-font:"Roboto Mono"}</style><link rel=stylesheet href=../../css/extra.css><script>__md_scope=new URL("../..",location),__md_hash=e=>[...e].reduce((e,_)=>(e<<5)-e+_.charCodeAt(0),0),__md_get=(e,_=localStorage,t=__md_scope)=>JSON.parse(_.getItem(t.pathname+"."+e)),__md_set=(e,_,t=localStorage,a=__md_scope)=>{try{t.setItem(a.pathname+"."+e,JSON.stringify(_))}catch(e){}}</script></head> <body dir=ltr data-md-color-scheme=default data-md-color-primary=indigo data-md-color-accent=indigo> <input class=md-toggle data-md-toggle=drawer type=checkbox id=__drawer autocomplete=off> <input class=md-toggle data-md-toggle=search type=checkbox id=__search autocomplete=off> <label class=md-overlay for=__drawer></label> <div data-md-component=skip> <a href=#output-files class=md-skip> Skip to content </a> </div> <div data-md-component=announce> </div> <div data-md-color-scheme=default data-md-component=outdated hidden> </div> <header class=md-header data-md-component=header> <nav class="md-header__inner md-grid" aria-label=Header> <a href=../.. title="XAVIER Documentation" class="md-header__button md-logo" aria-label="XAVIER Documentation" data-md-component=logo> <img src=../../assets/icons/doc-book.svg alt=logo> </a> <label class="md-header__button md-icon" for=__drawer> <svg xmlns=http://www.w3.org/2000/svg viewbox="0 0 24 24"><path d="M3 6h18v2H3V6m0 5h18v2H3v-2m0 5h18v2H3v-2Z"/></svg> </label> <div class=md-header__title data-md-component=header-title> <div class=md-header__ellipsis> <div class=md-header__topic> <span class=md-ellipsis> XAVIER Documentation </span> </div> <div class=md-header__topic data-md-component=header-topic> <span class=md-ellipsis> Output Files </span> </div> </div> </div> <form class=md-header__option data-md-component=palette> <input class=md-option data-md-color-media data-md-color-scheme=default data-md-color-primary=indigo data-md-color-accent=indigo aria-label="Switch to dark mode" type=radio name=__palette id=__palette_0> <label class="md-header__button md-icon" title="Switch to dark mode" for=__palette_1 hidden> <svg xmlns=http://www.w3.org/2000/svg viewbox="0 0 24 24"><path d="M17 6H7c-3.31 0-6 2.69-6 6s2.69 6 6 6h10c3.31 0 6-2.69 6-6s-2.69-6-6-6zm0 10H7c-2.21 0-4-1.79-4-4s1.79-4 4-4h10c2.21 0 4 1.79 4 4s-1.79 4-4 4zM7 9c-1.66 0-3 1.34-3 3s1.34 3 3 3 3-1.34 3-3-1.34-3-3-3z"/></svg> </label> <input class=md-option data-md-color-media data-md-color-scheme=slate data-md-color-primary=red data-md-color-accent=red aria-label="Switch to light mode" type=radio name=__palette id=__palette_1> <label class="md-header__button md-icon" title="Switch to light mode" for=__palette_0 hidden> <svg xmlns=http://www.w3.org/2000/svg viewbox="0 0 24 24"><path d="M17 7H7a5 5 0 0 0-5 5 5 5 0 0 0 5 5h10a5 5 0 0 0 5-5 5 5 0 0 0-5-5m0 8a3 3 0 0 1-3-3 3 3 0 0 1 3-3 3 3 0 0 1 3 3 3 3 0 0 1-3 3Z"/></svg> </label> </form> <script>var media,input,key,value,palette=__md_get("__palette");if(palette&&palette.color){"(prefers-color-scheme)"===palette.color.media&&(media=matchMedia("(prefers-color-scheme: light)"),input=document.querySelector(media.matches?"[data-md-color-media='(prefers-color-scheme: light)']":"[data-md-color-media='(prefers-color-scheme: dark)']"),palette.color.media=input.getAttribute("data-md-color-media"),palette.color.scheme=input.getAttribute("data-md-color-scheme"),palette.color.primary=input.getAttribute("data-md-color-primary"),palette.color.accent=input.getAttribute("data-md-color-accent"));for([key,value]of Object.entries(palette.color))document.body.setAttribute("data-md-color-"+key,value)}</script> <label class="md-header__button md-icon" for=__search> <svg xmlns=http://www.w3.org/2000/svg viewbox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.516 6.516 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5Z"/></svg> </label> <div class=md-search data-md-component=search role=dialog> <label class=md-search__overlay for=__search></label> <div class=md-search__inner role=search> <form class=md-search__form name=search> <input type=text class=md-search__input name=query aria-label=Search placeholder=Search autocapitalize=off autocorrect=off autocomplete=off spellcheck=false data-md-component=search-query required> <label class="md-search__icon md-icon" for=__search> <svg xmlns=http://www.w3.org/2000/svg viewbox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.516 6.516 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5Z"/></svg> <svg xmlns=http://www.w3.org/2000/svg viewbox="0 0 24 24"><path d="M20 11v2H8l5.5 5.5-1.42 1.42L4.16 12l7.92-7.92L13.5 5.5 8 11h12Z"/></svg> </label> <nav class=md-search__options aria-label=Search> <button type=reset class="md-search__icon md-icon" title=Clear aria-label=Clear tabindex=-1> <svg xmlns=http://www.w3.org/2000/svg viewbox="0 0 24 24"><path d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12 19 6.41Z"/></svg> </button> </nav> </form> <div class=md-search__output> <div class=md-search__scrollwrap data-md-scrollfix> <div class=md-search-result data-md-component=search-result> <div class=md-search-result__meta> Initializing search </div> <ol class=md-search-result__list role=presentation></ol> </div> </div> </div> </div> </div> <div class=md-header__source> <a href=https://github.com/CCBR/XAVIER title="Go to repository" class=md-source data-md-component=source> <div class="md-source__icon md-icon"> <svg xmlns=http://www.w3.org/2000/svg viewbox="0 0 448 512"><!-- Font Awesome Free 6.5.1 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2023 Fonticons, Inc.--><path d="M439.55 236.05 244 40.45a28.87 28.87 0 0 0-40.81 0l-40.66 40.63 51.52 51.52c27.06-9.14 52.68 16.77 43.39 43.68l49.66 49.66c34.23-11.8 61.18 31 35.47 56.69-26.49 26.49-70.21-2.87-56-37.34L240.22 199v121.85c25.3 12.54 22.26 41.85 9.08 55a34.34 34.34 0 0 1-48.55 0c-17.57-17.6-11.07-46.91 11.25-56v-123c-20.8-8.51-24.6-30.74-18.64-45L142.57 101 8.45 235.14a28.86 28.86 0 0 0 0 40.81l195.61 195.6a28.86 28.86 0 0 0 40.8 0l194.69-194.69a28.86 28.86 0 0 0 0-40.81z"/></svg> </div> <div class=md-source__repository> CCBR/XAVIER </div> </a> </div> </nav> </header> <div class=md-container data-md-component=container> <nav class=md-tabs aria-label=Tabs data-md-component=tabs> <div class=md-grid> <ul class=md-tabs__list> <li class=md-tabs__item> <a href=../.. class=md-tabs__link> About </a> </li> <li class=md-tabs__item> <a href=../../usage/run/ class=md-tabs__link> Command Line </a> </li> <li class=md-tabs__item> <a href=../../usage/gui/ class=md-tabs__link> Graphical Interface </a> </li> <li class="md-tabs__item md-tabs__item--active"> <a href=../overview/ class=md-tabs__link> Pipeline Details </a> </li> <li class=md-tabs__item> <a href=../../faq/questions/ class=md-tabs__link> FAQ </a> </li> <li class=md-tabs__item> <a href=../../license/ class=md-tabs__link> License </a> </li> </ul> </div> </nav> <main class=md-main data-md-component=main> <div class="md-main__inner md-grid"> <div class="md-sidebar md-sidebar--primary" data-md-component=sidebar data-md-type=navigation> <div class=md-sidebar__scrollwrap> <div class=md-sidebar__inner> <nav class="md-nav md-nav--primary md-nav--lifted md-nav--integrated" aria-label=Navigation data-md-level=0> <label class=md-nav__title for=__drawer> <a href=../.. title="XAVIER Documentation" class="md-nav__button md-logo" aria-label="XAVIER Documentation" data-md-component=logo> <img src=../../assets/icons/doc-book.svg alt=logo> </a> XAVIER Documentation </label> <div class=md-nav__source> <a href=https://github.com/CCBR/XAVIER title="Go to repository" class=md-source data-md-component=source> <div class="md-source__icon md-icon"> <svg xmlns=http://www.w3.org/2000/svg viewbox="0 0 448 512"><!-- Font Awesome Free 6.5.1 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2023 Fonticons, Inc.--><path d="M439.55 236.05 244 40.45a28.87 28.87 0 0 0-40.81 0l-40.66 40.63 51.52 51.52c27.06-9.14 52.68 16.77 43.39 43.68l49.66 49.66c34.23-11.8 61.18 31 35.47 56.69-26.49 26.49-70.21-2.87-56-37.34L240.22 199v121.85c25.3 12.54 22.26 41.85 9.08 55a34.34 34.34 0 0 1-48.55 0c-17.57-17.6-11.07-46.91 11.25-56v-123c-20.8-8.51-24.6-30.74-18.64-45L142.57 101 8.45 235.14a28.86 28.86 0 0 0 0 40.81l195.61 195.6a28.86 28.86 0 0 0 40.8 0l194.69-194.69a28.86 28.86 0 0 0 0-40.81z"/></svg> </div> <div class=md-source__repository> CCBR/XAVIER </div> </a> </div> <ul class=md-nav__list data-md-scrollfix> <li class=md-nav__item> <a href=../.. class=md-nav__link> <span class=md-ellipsis> About </span> </a> </li> <li class="md-nav__item md-nav__item--section md-nav__item--nested"> <input class="md-nav__toggle md-toggle " type=checkbox id=__nav_2> <label class=md-nav__link for=__nav_2 id=__nav_2_label tabindex> <span class=md-ellipsis> Command Line </span> <span class="md-nav__icon md-icon"></span> </label> <nav class=md-nav data-md-level=1 aria-labelledby=__nav_2_label aria-expanded=false> <label class=md-nav__title for=__nav_2> <span class="md-nav__icon md-icon"></span> Command Line </label> <ul class=md-nav__list data-md-scrollfix> <li class=md-nav__item> <a href=../../usage/run/ class=md-nav__link> <span class=md-ellipsis> xavier run </span> </a> </li> <li class=md-nav__item> <a href=../../usage/unlock/ class=md-nav__link> <span class=md-ellipsis> xavier unlock </span> </a> </li> <li class=md-nav__item> <a href=../../usage/cache/ class=md-nav__link> <span class=md-ellipsis> xavier cache </span> </a> </li> </ul> </nav> </li> <li class=md-nav__item> <a href=../../usage/gui/ class=md-nav__link> <span class=md-ellipsis> Graphical Interface </span> </a> </li> <li class="md-nav__item md-nav__item--active md-nav__item--section md-nav__item--nested"> <input class="md-nav__toggle md-toggle " type=checkbox id=__nav_4 checked> <label class=md-nav__link for=__nav_4 id=__nav_4_label tabindex> <span class=md-ellipsis> Pipeline Details </span> <span class="md-nav__icon md-icon"></span> </label> <nav class=md-nav data-md-level=1 aria-labelledby=__nav_4_label aria-expanded=true> <label class=md-nav__title for=__nav_4> <span class="md-nav__icon md-icon"></span> Pipeline Details </label> <ul class=md-nav__list data-md-scrollfix> <li class=md-nav__item> <a href=../overview/ class=md-nav__link> <span class=md-ellipsis> Overview </span> </a> </li> <li class=md-nav__item> <a href=../methods/ class=md-nav__link> <span class=md-ellipsis> Methods </span> </a> </li> <li class=md-nav__item> <a href=../tools/ class=md-nav__link> <span class=md-ellipsis> Tools </span> </a> </li> <li class=md-nav__item> <a href=../settings/ class=md-nav__link> <span class=md-ellipsis> Settings </span> </a> </li> <li class="md-nav__item md-nav__item--active"> <input class="md-nav__toggle md-toggle" type=checkbox id=__toc> <label class="md-nav__link md-nav__link--active" for=__toc> <span class=md-ellipsis> Output Files </span> <span class="md-nav__icon md-icon"></span> </label> <a href=./ class="md-nav__link md-nav__link--active"> <span class=md-ellipsis> Output Files </span> </a> <nav class="md-nav md-nav--secondary" aria-label="Table of contents"> <label class=md-nav__title for=__toc> <span class="md-nav__icon md-icon"></span> Table of contents </label> <ul class=md-nav__list data-md-component=toc data-md-scrollfix> <li class=md-nav__item> <a href=#xavier class=md-nav__link> <span class=md-ellipsis> XAVIER </span> </a> <nav class=md-nav aria-label=XAVIER> <ul class=md-nav__list> <li class=md-nav__item> <a href=#1-qc class=md-nav__link> <span class=md-ellipsis> 1. QC </span> </a> </li> <li class=md-nav__item> <a href=#2-bams class=md-nav__link> <span class=md-ellipsis> 2. bams </span> </a> </li> <li class=md-nav__item> <a href=#3-germline class=md-nav__link> <span class=md-ellipsis> 3. germline </span> </a> </li> <li class=md-nav__item> <a href=#4-logfiles class=md-nav__link> <span class=md-ellipsis> 4. logfiles </span> </a> </li> </ul> </nav> </li> <li class=md-nav__item> <a href=#tumor-normal-pair class=md-nav__link> <span class=md-ellipsis> Tumor-normal pair </span> </a> <nav class=md-nav aria-label="Tumor-normal pair"> <ul class=md-nav__list> <li class=md-nav__item> <a href=#somatic_paired class=md-nav__link> <span class=md-ellipsis> somatic_paired </span> </a> </li> </ul> </nav> </li> <li class=md-nav__item> <a href=#tumor-only class=md-nav__link> <span class=md-ellipsis> Tumor-only </span> </a> <nav class=md-nav aria-label=Tumor-only> <ul class=md-nav__list> <li class=md-nav__item> <a href=#somatic_tumor_only class=md-nav__link> <span class=md-ellipsis> somatic_tumor_only </span> </a> </li> </ul> </nav> </li> </ul> </nav> </li> </ul> </nav> </li> <li class="md-nav__item md-nav__item--section md-nav__item--nested"> <input class="md-nav__toggle md-toggle " type=checkbox id=__nav_5> <label class=md-nav__link for=__nav_5 id=__nav_5_label tabindex> <span class=md-ellipsis> FAQ </span> <span class="md-nav__icon md-icon"></span> </label> <nav class=md-nav data-md-level=1 aria-labelledby=__nav_5_label aria-expanded=false> <label class=md-nav__title for=__nav_5> <span class="md-nav__icon md-icon"></span> FAQ </label> <ul class=md-nav__list data-md-scrollfix> <li class=md-nav__item> <a href=../../faq/questions/ class=md-nav__link> <span class=md-ellipsis> General Questions </span> </a> </li> <li class=md-nav__item> <a href=../../contributing/ class=md-nav__link> <span class=md-ellipsis> How to contribute </span> </a> </li> </ul> </nav> </li> <li class=md-nav__item> <a href=../../license/ class=md-nav__link> <span class=md-ellipsis> License </span> </a> </li> </ul> </nav> </div> </div> </div> <div class=md-content data-md-component=content> <article class="md-content__inner md-typeset"> <h1 id=output-files>Output files<a class=headerlink href=#output-files title="Permanent link">&para;</a></h1> <h2 id=xavier>XAVIER<a class=headerlink href=#xavier title="Permanent link">&para;</a></h2> <p>The output files and their locations are broken down here for the XAVIER pipeline. Pre-processing and germline variant calling steps are common but somatic variant calling is dependent on whether the pipeline was run in either (A) tumor-normal pair or (B) tumor-only analysis mode. All file locations are relative to the output directory specified during the job submission.</p> <p>The output directory after a complete XAVIER run should look like:</p> <div class=highlight><pre><span></span><code>xavier_output/
├──<span class=w> </span>bams
├──<span class=w> </span>cluster.json<span class=w> </span><span class=c1># cluster info for the run</span>
├──<span class=w> </span>config
├──<span class=w> </span>config.json<span class=w>  </span><span class=c1># config file for the run</span>
├──<span class=w> </span>fastqs
├──<span class=w> </span>germline
├──<span class=w> </span>indels.vcf.gz<span class=o>[</span>.tbi<span class=o>]</span><span class=w> </span><span class=c1># raw germline INDELs</span>
├──<span class=w> </span>input_files
├──<span class=w> </span>intervals.list
├──<span class=w> </span><span class=o>{</span>sample1<span class=o>}</span>-normal.R1.fastq.gz<span class=w> </span>-&gt;<span class=w> </span>/path/to/<span class=o>{</span>sample1<span class=o>}</span>-normal.R1.fastq.gz
├──<span class=w> </span><span class=o>{</span>sample1<span class=o>}</span>-normal.R2.fastq.gz<span class=w> </span>-&gt;<span class=w> </span>/path/to/<span class=o>{</span>sample1<span class=o>}</span>-normal.R2.fastq.gz
├──<span class=w> </span><span class=o>{</span>sample1<span class=o>}</span>-tumor.R1.fastq.gz<span class=w> </span>-&gt;<span class=w> </span>/path/to/<span class=o>{</span>sample1<span class=o>}</span>-tumor.R1.fastq.gz
├──<span class=w> </span><span class=o>{</span>sample1<span class=o>}</span>-tumor.R2.fastq.gz<span class=w> </span>-&gt;<span class=w> </span>/path/to/<span class=o>{</span>sample1<span class=o>}</span>-tumor.R2.fastq.gz
.
.
.
├──<span class=w> </span>kickoff.sh
├──<span class=w> </span>logfiles
├──<span class=w> </span>QC
├──<span class=w> </span>resources
├──<span class=w> </span>snps.vcf.gz<span class=o>[</span>.tbi<span class=o>]</span><span class=w> </span><span class=c1># raw germline SNPs</span>
├──<span class=w> </span>somatic_paired<span class=w> </span><span class=c1># in case of tumor-normal paired run</span>
├──<span class=w> </span>somatic_tumor_only<span class=w> </span><span class=c1># in case of tumor-only run</span>
└──<span class=w> </span>workflow
</code></pre></div> <p>Below we describe the different folders that contain specific outputs obtained for all samples from the XAVIER pipeline</p> <h3 id=1-qc>1. <code>QC</code><a class=headerlink href=#1-qc title="Permanent link">&para;</a></h3> <p>The <code>QC</code> folder contains all the Quality-Control analyses performed at different steps of the pipeline for each sample to assess sequencing quality before and after adapter trimming, microbial taxonomic composition, contamination, variant calling, etc. The final summary report and data is available <code>finalQC</code> folder. \ The MultiQC report also contains results from other analysis like mapping statistics, ancestry and relatedness, etc. It is recommended to study the MultiQC report first to get a birds eye view of the sequence data quality.</p> <div class=highlight><pre><span></span><code>QC/
├──<span class=w> </span>exome_targets.bed
├──<span class=w> </span>finalQC/
│<span class=w>   </span>├──<span class=w> </span>MultiQC_Report_data
│<span class=w>   </span>└──<span class=w> </span>MultiQC_Report.html
├──<span class=w> </span>FQscreen
├──<span class=w> </span>sample1-normal
├──<span class=w> </span>sample1-normal_fastqc.html
.
.
.
├──<span class=w> </span>sample1-tumor
├──<span class=w> </span>sample1-tumor_fastqc.html
.
.
.
├──<span class=w> </span>kraken
├──<span class=w> </span>raw_variants.het
├──<span class=w> </span>raw_variants.variant_calling_detail_metrics
└──<span class=w> </span>raw_variants.variant_calling_summary_metrics
</code></pre></div> <h3 id=2-bams>2. <code>bams</code><a class=headerlink href=#2-bams title="Permanent link">&para;</a></h3> <p>The <code>bams</code> folder contain two subfolders <code>chrom_split</code> and <code>final_bams</code>. <code>final_bams</code> contains the final processed BAM files for each sample in the run and the <code>chrom_split</code> folder contains all the sample BAM files split by each chromosome.</p> <div class=highlight><pre><span></span><code>bams/
├──<span class=w> </span>chrom_split
│<span class=w>   </span>├──<span class=w> </span><span class=o>{</span>sample1<span class=o>}</span>-normal.chr1.split.bam<span class=o>[</span>.bai<span class=o>]</span>
│<span class=w>   </span>├──<span class=w> </span><span class=o>{</span>sample1<span class=o>}</span>-normal.chr2.split.bam<span class=o>[</span>.bai<span class=o>]</span>
.<span class=w>   </span>.
.<span class=w>   </span>.
.<span class=w>   </span>.
│<span class=w>   </span>└──<span class=w> </span><span class=o>{</span>sampleN<span class=o>}</span>-tumor.chrN.split.bam<span class=o>[</span>.bai<span class=o>]</span>
└──<span class=w> </span>final_bams
<span class=w>    </span>├──<span class=w> </span><span class=o>{</span>sample1<span class=o>}</span>-normal.bam<span class=o>[</span>.bai<span class=o>]</span>
<span class=w>    </span>├──<span class=w> </span><span class=o>{</span>sample1<span class=o>}</span>-tumor.bam<span class=o>[</span>.bai<span class=o>]</span>
<span class=w>    </span>.
<span class=w>    </span>.
<span class=w>    </span>.
<span class=w>    </span>└──<span class=w> </span><span class=o>{</span>sampleN<span class=o>}</span>-tumor.bam.bai
</code></pre></div> <h3 id=3-germline>3. <code>germline</code><a class=headerlink href=#3-germline title="Permanent link">&para;</a></h3> <p>This folder contains the output from the GATK Best Practices pipeline to obtain germline variants with a few alterations detailed below. Briefly, joint SNP and INDEL variant detection is conducted across all samples included in a pipeline run using the GATK Haplotypcaller under default settings. Raw variants are then subsequently filtered based on several GATK annotations: \ A strict set of criteria (QD &lt; 2.0, FS &gt; 60.0, MQ &lt; 40.0, MQRankSum &lt; -12.5, ReadPosRankSum &lt; -8.0 for SNPs; QD &lt; 2.0, FS &gt; 200.0, ReadPosRankSum &lt; -20.0 for INDELs) generates the 'combined.strictFilter.vcf'.</p> <p>This call set is highly stringent, maximizing the true positive rate at the expense of an elevated false negative rate. This call set is really only intended for more general population genetic scale analyses (e.g., burden tests, admixture, linkage/pedigree based analysis, etc.) where false positives can be significantly confounding.</p> <p>In case of human sequence data, a basic analyses of sample relatedness and ancestry (e.g., % European, African, etc.) is also performed using somalier.</p> <p>The output folder looks like:</p> <div class=highlight><pre><span></span><code>germline/
├──<span class=w> </span>gVCFs
.
.
.
├──<span class=w> </span>somalier<span class=w> </span><span class=c1># only for hg38 genome</span>
└──<span class=w> </span>VCF
</code></pre></div> <p>The <code>VCF</code> folder contains the final filtered germline variants (SNPs and INDELs) for all samples combined. The folder also contains raw variants for each sample, all samples combined, and also combined raw variants split by chromosome.</p> <div class=highlight><pre><span></span><code>VCF/
├──<span class=w> </span>by_chrom
│<span class=w>   </span>├──<span class=w> </span>raw_variants_byChrom.list
│<span class=w>   </span>├──<span class=w> </span>raw_variants.chr1.vcf.gz<span class=o>[</span>.tbi<span class=o>]</span>
.<span class=w>   </span>.
.<span class=w>   </span>.
.<span class=w>   </span>.
│<span class=w>   </span>└──<span class=w> </span>raw_variants.chrN.vcf.gz<span class=o>[</span>.tbi<span class=o>]</span>
├──<span class=w> </span>indel.filterd.vcf.gz<span class=o>[</span>.tbi<span class=o>]</span>
├──<span class=w> </span><span class=o>{</span>sample1<span class=o>}</span>-normal.germline.vcf.gz<span class=o>[</span>.tbi<span class=o>]</span>
├──<span class=w> </span><span class=o>{</span>sample1<span class=o>}</span>-tumor.germline.vcf.gz<span class=o>[</span>.tbi<span class=o>]</span>
.
.
.
├──<span class=w> </span><span class=o>{</span>sampleN<span class=o>}</span>-normal.germline.vcf.gz<span class=o>[</span>.tbi<span class=o>]</span>
├──<span class=w> </span><span class=o>{</span>sampleN<span class=o>}</span>-tumor.germline.vcf.gz<span class=o>[</span>.tbi<span class=o>]</span>
├──<span class=w> </span>raw_variants.vcf.gz<span class=o>[</span>.tbi<span class=o>]</span>
├──<span class=w> </span>snp.filtered.vcf.gz<span class=o>[</span>.tbi<span class=o>]</span>
└──<span class=w> </span>snp_indel.filtered.vcf.gz<span class=o>[</span>.tbi<span class=o>]</span>
</code></pre></div> <h3 id=4-logfiles>4. <code>logfiles</code><a class=headerlink href=#4-logfiles title="Permanent link">&para;</a></h3> <p>This folder contains the snakemake log files and computational statistics for the XAVIER run. All the log files (i.e., standard output and error) for each individual step are in the <code>slurmfiles</code> folder. These logfiles are important to diagnose errors in case the pipeline fails.</p> <div class=highlight><pre><span></span><code>logfiles/
├──<span class=w> </span>master.log
├──<span class=w> </span>mjobid.log
├──<span class=w> </span>runtime_statistics.json
├──<span class=w> </span>slurmfiles
├──<span class=w> </span>snakemake.log
├──<span class=w> </span>snakemake.log.jobby
└──<span class=w> </span>snakemake.log.jobby.short
</code></pre></div> <h2 id=tumor-normal-pair>Tumor-normal pair<a class=headerlink href=#tumor-normal-pair title="Permanent link">&para;</a></h2> <h3 id=somatic_paired><code>somatic_paired</code><a class=headerlink href=#somatic_paired title="Permanent link">&para;</a></h3> <p>This workflow calls somatic SNPs and INDELs using multiple variant detection algorithms. For each of these tools, variants are called in a paired tumor-normal fashion, with default settings. See <strong>Pipeline Details</strong> for more information about the tools used and their parameter settings.</p> <p>For each sample, the resulting VCF is fully annotated using VEP and converted to a MAF file using the vcf2maf tool. Resulting MAF files are found in <code>maf</code> folder within each caller's results directory (i.e., <code>mutect2_out</code>, <code>strelka_out</code>, etc.). Individual sample MAF files are then merged and saved in <code>merged_somatic_variants</code> directory.</p> <p>For Mutect2, we use a panel of normals (PON) developed from the ExAC (excluding TCGA) dataset, filtered for variants &lt;0.001 in the general population, and also including and in-house set of blacklisted recurrent germline variants that are not found in any population databases.</p> <p>For Copy Number Variants (CNVs), two tools are employed in tandem. First, Control-FREEC is run with default parameters. This generates pileup files that can be used by Sequenza, primarily for jointly estimating contamination and ploidy. These value are used to run Freec a second time for improved performance.</p> <p>The output directory should look like:</p> <div class=highlight><pre><span></span><code>somatic_paired/
├──<span class=w> </span>CNV<span class=w> </span><span class=c1># only if CNVs analyzed</span>
│<span class=w>   </span>├──<span class=w> </span>freec_out
│<span class=w>   </span>└──<span class=w> </span>sequenza_out
├──<span class=w> </span>ffpe_filter<span class=w> </span><span class=c1># only if FFPE filter applied</span>
├──<span class=w> </span>qc
└──<span class=w> </span>SNP_Indels
<span class=w>    </span>├──<span class=w> </span>merged_somatic_variants
<span class=w>    </span>│<span class=w>   </span>├──<span class=w> </span>cohort_summary
<span class=w>    </span>│<span class=w>   </span>├──<span class=w> </span>maf<span class=w> </span><span class=c1># Final merged MAFs for each sample</span>
<span class=w>    </span>│<span class=w>   </span>└──<span class=w> </span>vcf
<span class=w>    </span>├──<span class=w> </span>mutect2_out
<span class=w>    </span>│<span class=w>   </span>├──<span class=w> </span>chrom_split
<span class=w>    </span>│<span class=w>   </span>├──<span class=w> </span>cohort_summary
<span class=w>    </span>│<span class=w>   </span>├──<span class=w> </span>maf
<span class=w>    </span>│<span class=w>   </span>├──<span class=w> </span>pileup_summaries
<span class=w>    </span>│<span class=w>   </span>├──<span class=w> </span>read_orientation_data
<span class=w>    </span>│<span class=w>   </span>└──<span class=w> </span>vcf
<span class=w>    </span>├──<span class=w> </span>mutect_out
<span class=w>    </span>│<span class=w>   </span>├──<span class=w> </span>maf
<span class=w>    </span>│<span class=w>   </span>├──<span class=w> </span>.
<span class=w>    </span>│<span class=w>   </span>├──<span class=w> </span>.
<span class=w>    </span>│<span class=w>   </span>└──<span class=w> </span>vcf
<span class=w>    </span>├──<span class=w> </span>strelka_out
<span class=w>    </span>├──<span class=w> </span>vardict_out
<span class=w>    </span>└──<span class=w> </span>varscan_out
</code></pre></div> <h2 id=tumor-only>Tumor-only<a class=headerlink href=#tumor-only title="Permanent link">&para;</a></h2> <h3 id=somatic_tumor_only><code>somatic_tumor_only</code><a class=headerlink href=#somatic_tumor_only title="Permanent link">&para;</a></h3> <p>In general, the tumor-only pipeline is a stripped down version of the tumor-normal pipeline. We only run MuTect2, Mutect, and VarDict for somatic variant detection, with the same PON and filtering as described above for the tumor-normal pipeline.</p> <div class=highlight><pre><span></span><code>somatic_tumor_only/
├──<span class=w> </span>ffpe_filter<span class=w> </span><span class=c1># only if FFPE filter applied</span>
├──<span class=w> </span>qc
└──<span class=w> </span>SNP_Indels
<span class=w>    </span>├──<span class=w> </span>merged_somatic_variants
<span class=w>    </span>│<span class=w>   </span>├──<span class=w> </span>cohort_summary
<span class=w>    </span>│<span class=w>   </span>├──<span class=w> </span>maf<span class=w> </span><span class=c1># Final merged MAFs for each sample</span>
<span class=w>    </span>│<span class=w>   </span>└──<span class=w> </span>vcf
<span class=w>    </span>├──<span class=w> </span>mutect2_out
<span class=w>    </span>│<span class=w>   </span>├──<span class=w> </span>chrom_split
<span class=w>    </span>│<span class=w>   </span>├──<span class=w> </span>cohort_summary
<span class=w>    </span>│<span class=w>   </span>├──<span class=w> </span>maf
<span class=w>    </span>│<span class=w>   </span>├──<span class=w> </span>read_orientation_data
<span class=w>    </span>│<span class=w>   </span>└──<span class=w> </span>vcf
<span class=w>    </span>├──<span class=w> </span>mutect_out
<span class=w>    </span>│<span class=w>   </span>├──<span class=w> </span>maf
<span class=w>    </span>│<span class=w>   </span>├──<span class=w> </span>.
<span class=w>    </span>│<span class=w>   </span>├──<span class=w> </span>.
<span class=w>    </span>│<span class=w>   </span>└──<span class=w> </span>vcf
<span class=w>    </span>├──<span class=w> </span>vardict_out
<span class=w>    </span>└──<span class=w> </span>varscan_out
</code></pre></div> <aside class=md-source-file> <span class=md-source-file__fact> <span class=md-icon title="Last update"> <svg xmlns=http://www.w3.org/2000/svg viewbox="0 0 24 24"><path d="M21 13.1c-.1 0-.3.1-.4.2l-1 1 2.1 2.1 1-1c.2-.2.2-.6 0-.8l-1.3-1.3c-.1-.1-.2-.2-.4-.2m-1.9 1.8-6.1 6V23h2.1l6.1-6.1-2.1-2M12.5 7v5.2l4 2.4-1 1L11 13V7h1.5M11 21.9c-5.1-.5-9-4.8-9-9.9C2 6.5 6.5 2 12 2c5.3 0 9.6 4.1 10 9.3-.3-.1-.6-.2-1-.2s-.7.1-1 .2C19.6 7.2 16.2 4 12 4c-4.4 0-8 3.6-8 8 0 4.1 3.1 7.5 7.1 7.9l-.1.2v1.8Z"/></svg> </span> 2024-01-30 </span> </aside> </article> </div> <script>var target=document.getElementById(location.hash.slice(1));target&&target.name&&(target.checked=target.name.startsWith("__tabbed_"))</script> </div> <button type=button class="md-top md-icon" data-md-component=top hidden> <svg xmlns=http://www.w3.org/2000/svg viewbox="0 0 24 24"><path d="M13 20h-2V8l-5.5 5.5-1.42-1.42L12 4.16l7.92 7.92-1.42 1.42L13 8v12Z"/></svg> Back to top </button> </main> <footer class=md-footer> <div class="md-footer-meta md-typeset"> <div class="md-footer-meta__inner md-grid"> <div class=md-copyright> <div class=md-copyright__highlight> Copyright &copy; 2023 CCBR </div> Made with <a href=https://squidfunk.github.io/mkdocs-material/ target=_blank rel=noopener> Material for MkDocs </a> </div> <div class=md-social> <a href=http://bioinformatics.cancer.gov target=_blank rel=noopener title=bioinformatics.cancer.gov class=md-social__link> <svg xmlns=http://www.w3.org/2000/svg viewbox="0 0 640 512"><!-- Font Awesome Free 6.5.1 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2023 Fonticons, Inc.--><path d="M144 0a80 80 0 1 1 0 160 80 80 0 1 1 0-160zm368 0a80 80 0 1 1 0 160 80 80 0 1 1 0-160zM0 298.7C0 239.8 47.8 192 106.7 192h42.7c15.9 0 31 3.5 44.6 9.7-1.3 7.2-1.9 14.7-1.9 22.3 0 38.2 16.8 72.5 43.3 96H21.3C9.6 320 0 310.4 0 298.7zM405.3 320h-.7c26.6-23.5 43.3-57.8 43.3-96 0-7.6-.7-15-1.9-22.3 13.6-6.3 28.7-9.7 44.6-9.7h42.7c58.9 0 106.7 47.8 106.7 106.7 0 11.8-9.6 21.3-21.3 21.3H405.3zM224 224a96 96 0 1 1 192 0 96 96 0 1 1-192 0zm-96 261.3c0-73.6 59.7-133.3 133.3-133.3h117.4c73.6 0 133.3 59.7 133.3 133.3 0 14.7-11.9 26.7-26.7 26.7H154.7c-14.7 0-26.7-11.9-26.7-26.7z"/></svg> </a> <a href=https://github.com/CCBR target=_blank rel=noopener title=github.com class=md-social__link> <svg xmlns=http://www.w3.org/2000/svg viewbox="0 0 496 512"><!-- Font Awesome Free 6.5.1 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2023 Fonticons, Inc.--><path d="M165.9 397.4c0 2-2.3 3.6-5.2 3.6-3.3.3-5.6-1.3-5.6-3.6 0-2 2.3-3.6 5.2-3.6 3-.3 5.6 1.3 5.6 3.6zm-31.1-4.5c-.7 2 1.3 4.3 4.3 4.9 2.6 1 5.6 0 6.2-2s-1.3-4.3-4.3-5.2c-2.6-.7-5.5.3-6.2 2.3zm44.2-1.7c-2.9.7-4.9 2.6-4.6 4.9.3 2 2.9 3.3 5.9 2.6 2.9-.7 4.9-2.6 4.6-4.6-.3-1.9-3-3.2-5.9-2.9zM244.8 8C106.1 8 0 113.3 0 252c0 110.9 69.8 205.8 169.5 239.2 12.8 2.3 17.3-5.6 17.3-12.1 0-6.2-.3-40.4-.3-61.4 0 0-70 15-84.7-29.8 0 0-11.4-29.1-27.8-36.6 0 0-22.9-15.7 1.6-15.4 0 0 24.9 2 38.6 25.8 21.9 38.6 58.6 27.5 72.9 20.9 2.3-16 8.8-27.1 16-33.7-55.9-6.2-112.3-14.3-112.3-110.5 0-27.5 7.6-41.3 23.6-58.9-2.6-6.5-11.1-33.3 2.6-67.9 20.9-6.5 69 27 69 27 20-5.6 41.5-8.5 62.8-8.5s42.8 2.9 62.8 8.5c0 0 48.1-33.6 69-27 13.7 34.7 5.2 61.4 2.6 67.9 16 17.7 25.8 31.5 25.8 58.9 0 96.5-58.9 104.2-114.8 110.5 9.2 7.9 17 22.9 17 46.4 0 33.7-.3 75.4-.3 83.6 0 6.5 4.6 14.4 17.3 12.1C428.2 457.8 496 362.9 496 252 496 113.3 383.5 8 244.8 8zM97.2 352.9c-1.3 1-1 3.3.7 5.2 1.6 1.6 3.9 2.3 5.2 1 1.3-1 1-3.3-.7-5.2-1.6-1.6-3.9-2.3-5.2-1zm-10.8-8.1c-.7 1.3.3 2.9 2.3 3.9 1.6 1 3.6.7 4.3-.7.7-1.3-.3-2.9-2.3-3.9-2-.6-3.6-.3-4.3.7zm32.4 35.6c-1.6 1.3-1 4.3 1.3 6.2 2.3 2.3 5.2 2.6 6.5 1 1.3-1.3.7-4.3-1.3-6.2-2.2-2.3-5.2-2.6-6.5-1zm-11.4-14.7c-1.6 1-1.6 3.6 0 5.9 1.6 2.3 4.3 3.3 5.6 2.3 1.6-1.3 1.6-3.9 0-6.2-1.4-2.3-4-3.3-5.6-2z"/></svg> </a> <a href=https://hub.docker.com/orgs/nciccbr/repositories target=_blank rel=noopener title=hub.docker.com class=md-social__link> <svg xmlns=http://www.w3.org/2000/svg viewbox="0 0 640 512"><!-- Font Awesome Free 6.5.1 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2023 Fonticons, Inc.--><path d="M349.9 236.3h-66.1v-59.4h66.1v59.4zm0-204.3h-66.1v60.7h66.1V32zm78.2 144.8H362v59.4h66.1v-59.4zm-156.3-72.1h-66.1v60.1h66.1v-60.1zm78.1 0h-66.1v60.1h66.1v-60.1zm276.8 100c-14.4-9.7-47.6-13.2-73.1-8.4-3.3-24-16.7-44.9-41.1-63.7l-14-9.3-9.3 14c-18.4 27.8-23.4 73.6-3.7 103.8-8.7 4.7-25.8 11.1-48.4 10.7H2.4c-8.7 50.8 5.8 116.8 44 162.1 37.1 43.9 92.7 66.2 165.4 66.2 157.4 0 273.9-72.5 328.4-204.2 21.4.4 67.6.1 91.3-45.2 1.5-2.5 6.6-13.2 8.5-17.1l-13.3-8.9zm-511.1-27.9h-66v59.4h66.1v-59.4zm78.1 0h-66.1v59.4h66.1v-59.4zm78.1 0h-66.1v59.4h66.1v-59.4zm-78.1-72.1h-66.1v60.1h66.1v-60.1z"/></svg> </a> </div> </div> </div> </footer> </div> <div class=md-dialog data-md-component=dialog> <div class="md-dialog__inner md-typeset"></div> </div> <script id=__config type=application/json>{"base": "../..", "features": ["navigation.tabs", "navigation.top", "toc.integrate"], "search": "../../assets/javascripts/workers/search.b8dbb3d2.min.js", "translations": {"clipboard.copied": "Copied to clipboard", "clipboard.copy": "Copy to clipboard", "search.result.more.one": "1 more on this page", "search.result.more.other": "# more on this page", "search.result.none": "No matching documents", "search.result.one": "1 matching document", "search.result.other": "# matching documents", "search.result.placeholder": "Type to start searching", "search.result.term.missing": "Missing", "select.version": "Select version"}, "version": {"provider": "mike"}}</script> <script src=../../assets/javascripts/bundle.e1c3ead8.min.js></script> </body> </html>