<!DOCTYPE html><html class=writer-html5 lang=en> <head><meta charset=utf-8><meta http-equiv=X-UA-Compatible content="IE=edge"><meta name=viewport content="width=device-width, initial-scale=1.0"><meta name=author content=CCBR><link rel="shortcut icon" href=../../img/favicon.ico><title>Methods - XAVIER Documentation</title><link rel=stylesheet href=../../css/theme.css><link rel=stylesheet href=../../css/theme_extra.css><link rel=stylesheet href=https://cdnjs.cloudflare.com/ajax/libs/highlight.js/11.8.0/styles/github.min.css><link href=../../css/custom.css rel=stylesheet><link href=../../css/version-select.css rel=stylesheet><script>
        // Current page data
        var mkdocs_page_name = "Methods";
        var mkdocs_page_input_path = "pipeline-details/methods.md";
        var mkdocs_page_url = null;
      </script><!--[if lt IE 9]>
      <script src="../../js/html5shiv.min.js"></script>
    <![endif]--><script src=https://cdnjs.cloudflare.com/ajax/libs/highlight.js/11.8.0/highlight.min.js></script><script>hljs.highlightAll();</script></head> <body class=wy-body-for-nav role=document> <div class=wy-grid-for-nav> <nav data-toggle=wy-nav-shift class="wy-nav-side stickynav"> <div class=wy-side-scroll> <div class=wy-side-nav-search> <a href=../.. class="icon icon-home"> XAVIER Documentation </a><div role=search> <form id=rtd-search-form class=wy-form action=../../search.html method=get> <input type=text name=q placeholder="Search docs" aria-label="Search docs" title="Type search term here"> </form> </div> </div> <div class="wy-menu wy-menu-vertical" data-spy=affix role=navigation aria-label="Navigation menu"> <ul> <li class=toctree-l1><a class="reference internal" href=../..>About</a> </li> </ul> <p class=caption><span class=caption-text>Command Line</span></p> <ul> <li class=toctree-l1><a href=../../usage/run/ class="reference internal">xavier run</a> </li> <li class=toctree-l1><a href=../../usage/unlock/ class="reference internal">xavier unlock</a> </li> <li class=toctree-l1><a href=../../usage/cache/ class="reference internal">xavier cache</a> </li> </ul> <ul> <li class=toctree-l1><a href=../../usage/gui/ class="reference internal">Graphical Interface</a> </li> </ul> <p class=caption><span class=caption-text>Pipeline Details</span></p> <ul class=current> <li class=toctree-l1><a href=../overview/ class="reference internal">Overview</a> </li> <li class="toctree-l1 current"><a class="reference internal current" href=#>Methods</a> <ul class=current> <li class=toctree-l2><a class="reference internal" href=#data-preprocessing>Data preprocessing</a> </li> <li class=toctree-l2><a class="reference internal" href=#germline-variant-calling>Germline variant calling</a> </li> <li class=toctree-l2><a class="reference internal" href=#somatic-variant-calling>Somatic variant calling</a> </li> <li class=toctree-l2><a class="reference internal" href=#ffpe-artifact-filtering>FFPE Artifact filtering</a> </li> <li class=toctree-l2><a class="reference internal" href=#quality-and-identity-metrics>Quality and identity metrics</a> </li> <li class=toctree-l2><a class="reference internal" href=#pipeline-orchestration>Pipeline Orchestration</a> </li> <li class=toctree-l2><a class="reference internal" href=#references>References</a> </li> </ul> </li> <li class=toctree-l1><a href=../tools/ class="reference internal">Tools</a> </li> <li class=toctree-l1><a href=../settings/ class="reference internal">Settings</a> </li> <li class=toctree-l1><a href=../output/ class="reference internal">Output Files</a> </li> </ul> <p class=caption><span class=caption-text>FAQ</span></p> <ul> <li class=toctree-l1><a href=../../faq/questions/ class="reference internal">General Questions</a> </li> <li class=toctree-l1><a href=../../contributing/ class="reference internal">How to contribute</a> </li> </ul> <ul> <li class=toctree-l1><a href=../../license/ class="reference internal">License</a> </li> </ul> </div> </div> </nav> <section data-toggle=wy-nav-shift class=wy-nav-content-wrap> <nav class=wy-nav-top role=navigation aria-label="Mobile navigation menu"> <i data-toggle=wy-nav-top class="fa fa-bars"></i> <a href=../..>XAVIER Documentation</a> </nav> <div class=wy-nav-content> <div class=rst-content><div role=navigation aria-label="breadcrumbs navigation"> <ul class=wy-breadcrumbs> <li><a href=../.. class="icon icon-home" aria-label=Docs></a></li> <li class=breadcrumb-item>Pipeline Details</li> <li class="breadcrumb-item active">Methods</li> <li class=wy-breadcrumbs-aside> <a href=https://github.com/CCBR/XAVIER/edit/unified/docs/pipeline-details/methods.md>Edit on CCBR/XAVIER</a> </li> </ul> <hr> </div> <div role=main class=document itemscope=itemscope itemtype=http://schema.org/Article> <div class=section itemprop=articleBody> <h1 id=methods-description>Methods description<a class=headerlink href=#methods-description title="Permanent link">&para;</a></h1> <p>This page contains a description of all methods used in the pipeline, along with references for important tools.</p> <p><strong>Note that depending on the settings used, not all of these methods may be applicable, so please adapt this text appropriately for your application.</strong></p> <p>You can also download this text as a Word document (.docx) that contains an EndNote traveling library using the button below.</p> <p><a href=../assets/wes_pipeline_methods.docx><img alt="Download DOCX" src="https://custom-icon-badges.herokuapp.com/badge/-Download-blue?style=for-the-badge&logo=download&logoColor=white" title="Download Methods Document"></a></p> <hr> <h2 id=data-preprocessing>Data preprocessing<a class=headerlink href=#data-preprocessing title="Permanent link">&para;</a></h2> <p>Low-quality and adapters sequences are trimmed from the raw sequencing reads using Trimmomatic (v. 0.39)<sup id=fnref:1><a class=footnote-ref href=#fn:1>1</a></sup>. Trimmed reads are then aligned to the human hg38 reference genome using BWA mapping software (v. 0.7.17)<sup id=fnref:2><a class=footnote-ref href=#fn:2>2</a></sup>. Duplicate reads are marked using Samblaster (v. 0.1.25)<sup id=fnref:3><a class=footnote-ref href=#fn:3>3</a></sup> and sorted using samtools (v. 1.8). Finally, base quality score recalibration is performed as indicated in the GATK4 (v. 4.2.2.0) best practices <sup id=fnref:4><a class=footnote-ref href=#fn:4>4</a></sup>.</p> <h2 id=germline-variant-calling>Germline variant calling<a class=headerlink href=#germline-variant-calling title="Permanent link">&para;</a></h2> <p>HaplotypeCaller from GATK4 (v. 4.2.2.0) is used to call germline variants, parallelized across chromosomes, and all samples in the cohort are joint genotyped together <sup id=fnref2:4><a class=footnote-ref href=#fn:4>4</a></sup><sup>,</sup><sup id=fnref:5><a class=footnote-ref href=#fn:5>5</a></sup>.</p> <h2 id=somatic-variant-calling>Somatic variant calling<a class=headerlink href=#somatic-variant-calling title="Permanent link">&para;</a></h2> <p>Somatic variant calling (SNPs and Indels) is performed using Mutect (v. 1.1.7)<sup id=fnref:6><a class=footnote-ref href=#fn:6>6</a></sup>, Mutect2 (GATK v. 4.2.0)<sup id=fnref:7><a class=footnote-ref href=#fn:7>7</a></sup>, Strelka2 (v. 2.9.0)<sup id=fnref:8><a class=footnote-ref href=#fn:8>8</a></sup>, and VarDict (v. 1.4)<sup id=fnref:9><a class=footnote-ref href=#fn:9>9</a></sup> in tumor-normal mode. Variants from all callers are merged using the CombineVariants tool from GATK version 3.8-1. Genomic, functional and consequence annotations are added using Variant Effect Predictor (VEP v. 99)<sup id=fnref:10><a class=footnote-ref href=#fn:10>10</a></sup> and converted to Mutation Annotation Format (MAF) using the vcf2maf tool (v. 1.6.16)<sup id=fnref:11><a class=footnote-ref href=#fn:11>11</a></sup>.</p> <p>For Copy Number Variants (CNVs), Control-Freec (v. 11.6)<sup id=fnref:12><a class=footnote-ref href=#fn:12>12</a></sup> is used to generate pileups, which are used as input for the R package 'sequenza' (v. 3.0.0)<sup id=fnref:13><a class=footnote-ref href=#fn:13>13</a></sup>. The complete Control-Freec workflow is then re-run using ploidy and cellularity estimates from 'sequenza'.</p> <h2 id=ffpe-artifact-filtering>FFPE Artifact filtering<a class=headerlink href=#ffpe-artifact-filtering title="Permanent link">&para;</a></h2> <p>SOBDetector is a tool that scores variants based on strand-orientation bias, which can be a sign of DNA damage caused by fixation of tissue. This pipeline runs SOBDetector in a two-pass method. The first pass uses parameters provided with the software (calculated from publicly available data from TCGA), then cohort-specific bias metrics are computed from those results, and SOBDetector is re-run using these cohort-specific values.</p> <h2 id=quality-and-identity-metrics>Quality and identity metrics<a class=headerlink href=#quality-and-identity-metrics title="Permanent link">&para;</a></h2> <p>Ancestry and relatedness scores are generated using Somalier (v. 0.2.13)<sup id=fnref:14><a class=footnote-ref href=#fn:14>14</a></sup>. Contamination analyses are performed against viral and bacterial genomes from NCBI using Kraken2 (v. 2.1.2)<sup id=fnref:15><a class=footnote-ref href=#fn:15>15</a></sup>, as well as against mouse, human, and UniVec databases using FastQ Screen (v. 0.14.1)<sup id=fnref:16><a class=footnote-ref href=#fn:16>16</a></sup>. Sequence, mapping and variant statistics are computed using FastQC (v. 0.11.9), Qualimap (v. 2.2.1)<sup id=fnref:17><a class=footnote-ref href=#fn:17>17</a></sup> and SNPeff (v. 4.3t)<sup id=fnref:18><a class=footnote-ref href=#fn:18>18</a></sup>. All of these metrics are combined into an interactive HTML report using MultiQC (v. 1.11)<sup id=fnref:19><a class=footnote-ref href=#fn:19>19</a></sup>.</p> <h2 id=pipeline-orchestration>Pipeline Orchestration<a class=headerlink href=#pipeline-orchestration title="Permanent link">&para;</a></h2> <p>Job execution and management is done using Snakemake (v. 6.8.2)<sup id=fnref:20><a class=footnote-ref href=#fn:20>20</a></sup> using custom-built Singularity (v. 3.8.5) containers for reproducibility.</p> <h2 id=references>References<a class=headerlink href=#references title="Permanent link">&para;</a></h2> <div class=footnote> <hr> <ol> <li id=fn:1> <p>Bolger, A.M., M. Lohse, and B. Usadel, Trimmomatic: a flexible trimmer for Illumina sequence data. Bioinformatics, 2014. 30(15): p. 2114-20.&#160;<a class=footnote-backref href=#fnref:1 title="Jump back to footnote 1 in the text">&#8617;</a></p> </li> <li id=fn:2> <p>Li, H. and R. Durbin, Fast and accurate short read alignment with Burrows-Wheeler transform. Bioinformatics, 2009. 25(14): p. 1754-60.&#160;<a class=footnote-backref href=#fnref:2 title="Jump back to footnote 2 in the text">&#8617;</a></p> </li> <li id=fn:3> <p>Faust, G.G. and I.M. Hall, SAMBLASTER: fast duplicate marking and structural variant read extraction. Bioinformatics, 2014. 30(17): p. 2503-5.&#160;<a class=footnote-backref href=#fnref:3 title="Jump back to footnote 3 in the text">&#8617;</a></p> </li> <li id=fn:4> <p>Van der Auwera, G.A. and B.D. O'Connor, Genomics in the cloud : using Docker, GATK, and WDL in Terra. First edition. ed. 2020, Sebastopol, CA: O'Reilly Media.&#160;<a class=footnote-backref href=#fnref:4 title="Jump back to footnote 4 in the text">&#8617;</a><a class=footnote-backref href=#fnref2:4 title="Jump back to footnote 4 in the text">&#8617;</a></p> </li> <li id=fn:5> <p>Poplin, R., et al., Scaling accurate genetic variant discovery to tens of thousands of samples. bioRxiv, 2018: p. 201178.&#160;<a class=footnote-backref href=#fnref:5 title="Jump back to footnote 5 in the text">&#8617;</a></p> </li> <li id=fn:6> <p>Cibulskis, K., et al., Sensitive detection of somatic point mutations in impure and heterogeneous cancer samples. Nat Biotechnol, 2013. 31(3): p. 213-9.&#160;<a class=footnote-backref href=#fnref:6 title="Jump back to footnote 6 in the text">&#8617;</a></p> </li> <li id=fn:7> <p>Benjamin, D., et al., Calling Somatic SNVs and Indels with Mutect2. bioRxiv, 2019: p. 861054.&#160;<a class=footnote-backref href=#fnref:7 title="Jump back to footnote 7 in the text">&#8617;</a></p> </li> <li id=fn:8> <p>Kim, S., et al., Strelka2: fast and accurate calling of germline and somatic variants. Nat Methods, 2018. 15(8): p. 591-594.&#160;<a class=footnote-backref href=#fnref:8 title="Jump back to footnote 8 in the text">&#8617;</a></p> </li> <li id=fn:9> <p>Lai, Z., et al., VarDict: a novel and versatile variant caller for next-generation sequencing in cancer research. Nucleic Acids Res, 2016. 44(11): p. e108.&#160;<a class=footnote-backref href=#fnref:9 title="Jump back to footnote 9 in the text">&#8617;</a></p> </li> <li id=fn:10> <p>McLaren, W., et al., The Ensembl Variant Effect Predictor. Genome Biol, 2016. 17(1): p. 122.&#160;<a class=footnote-backref href=#fnref:10 title="Jump back to footnote 10 in the text">&#8617;</a></p> </li> <li id=fn:11> <p>Memorial Sloan Kettering Cancer Center. vcf2maf. 2013; Available from: https://github.com/mskcc/vcf2maf.&#160;<a class=footnote-backref href=#fnref:11 title="Jump back to footnote 11 in the text">&#8617;</a></p> </li> <li id=fn:12> <p>Boeva, V., et al., Control-FREEC: a tool for assessing copy number and allelic content using next-generation sequencing data. Bioinformatics, 2012. 28(3): p. 423-5.&#160;<a class=footnote-backref href=#fnref:12 title="Jump back to footnote 12 in the text">&#8617;</a></p> </li> <li id=fn:13> <p>Favero, F., et al., Sequenza: allele-specific copy number and mutation profiles from tumor sequencing data. Ann Oncol, 2015. 26(1): p. 64-70.&#160;<a class=footnote-backref href=#fnref:13 title="Jump back to footnote 13 in the text">&#8617;</a></p> </li> <li id=fn:14> <p>Pedersen, B. somalier: extract informative sites, evaluate relatedness, and perform quality-control on BAM/CRAM/BCF/VCF/GVCF. 2018; Available from: https://github.com/brentp/somalier.&#160;<a class=footnote-backref href=#fnref:14 title="Jump back to footnote 14 in the text">&#8617;</a></p> </li> <li id=fn:15> <p>Wood, D.E., J. Lu, and B. Langmead, Improved metagenomic analysis with Kraken 2. Genome Biol, 2019. 20(1): p. 257.&#160;<a class=footnote-backref href=#fnref:15 title="Jump back to footnote 15 in the text">&#8617;</a></p> </li> <li id=fn:16> <p>Wingett, S.W. and S. Andrews, FastQ Screen: A tool for multi-genome mapping and quality control. F1000Res, 2018. 7: p. 1338.&#160;<a class=footnote-backref href=#fnref:16 title="Jump back to footnote 16 in the text">&#8617;</a></p> </li> <li id=fn:17> <p>Okonechnikov, K., A. Conesa, and F. Garcia-Alcalde, Qualimap 2: advanced multi-sample quality control for high-throughput sequencing data. Bioinformatics, 2016. 32(2): p. 292-4.&#160;<a class=footnote-backref href=#fnref:17 title="Jump back to footnote 17 in the text">&#8617;</a></p> </li> <li id=fn:18> <p>Cingolani, P., et al., A program for annotating and predicting the effects of single nucleotide polymorphisms, SnpEff: SNPs in the genome of Drosophila melanogaster strain w1118; iso-2; iso-3. Fly (Austin), 2012. 6(2): p. 80-92.&#160;<a class=footnote-backref href=#fnref:18 title="Jump back to footnote 18 in the text">&#8617;</a></p> </li> <li id=fn:19> <p>Ewels, P., et al., MultiQC: summarize analysis results for multiple tools and samples in a single report. Bioinformatics, 2016. 32(19): p. 3047-8.&#160;<a class=footnote-backref href=#fnref:19 title="Jump back to footnote 19 in the text">&#8617;</a></p> </li> <li id=fn:20> <p>Koster, J. and S. Rahmann, Snakemake-a scalable bioinformatics workflow engine. Bioinformatics, 2018. 34(20): p. 3600.&#160;<a class=footnote-backref href=#fnref:20 title="Jump back to footnote 20 in the text">&#8617;</a></p> </li> </ol> </div> </div> </div><script async src="https://www.googletagmanager.com/gtag/js?id=G-D3SL9V30KL"></script> <script>
  window.dataLayer = window.dataLayer || [];
  function gtag(){dataLayer.push(arguments);}
  gtag('js', new Date());
  gtag('config', 'G-D3SL9V30KL'); // Replace with your tracking ID
</script> </div> </div> </section> </div> <div class=rst-versions role=note aria-label=Versions> <span class=rst-current-version data-toggle=rst-current-version> <span> <a href=https://github.com/CCBR/XAVIER class="fa fa-code-fork" style="color: #fcfcfc"> CCBR/XAVIER</a> </span> <span><a href=../overview/ style="color: #fcfcfc">&laquo; Previous</a></span> <span><a href=../tools/ style="color: #fcfcfc">Next &raquo;</a></span> </span> </div> <script src=../../js/jquery-3.6.0.min.js></script> <script>var base_url = "../..";</script> <script src=../../js/theme_extra.js></script> <script src=../../js/theme.js></script> <script src=../../js/custom.js></script> <script src=../../search/main.js></script> <script src=../../js/version-select.js></script> <script>
        jQuery(function () {
            SphinxRtdTheme.Navigation.enable(true);
        });
    </script> </body> </html> 